Index: programs/Xserver/hw/xfree86/drivers/i810/i810_driver.c
===================================================================
RCS file: /cvs/xorg/xc/programs/Xserver/hw/xfree86/drivers/i810/i810_driver.c,v
retrieving revision 1.10
diff -u -p -r1.10 i810_driver.c
--- programs/Xserver/hw/xfree86/drivers/i810/i810_driver.c	17 Sep 2004 03:04:51 -0000	1.10
+++ programs/Xserver/hw/xfree86/drivers/i810/i810_driver.c	29 Sep 2004 15:44:35 -0000
@@ -903,12 +903,15 @@ I810PreInit(ScrnInfoPtr pScrn, int flags
     * size in the XF86Config, use that amount instead.
     *
     *  Changed to 8 Meg so we can have acceleration by default (Mark).
+    *  16 MB is not enough for 3D in 1600x1200 (Matthias)
     */
    mem = I810CheckAvailableMemory(pScrn);
-   if (pI810->directRenderingDisabled || mem < 134217728)  /* < 128 MB */
+   if (pI810->directRenderingDisabled || mem < 131072)       /* < 128 MB */
        pScrn->videoRam = 8192;
-   else
+   else if (mem < 196608)  /* < 192 MB */
        pScrn->videoRam = 16384;
+   else
+       pScrn->videoRam = 24576;
    from = X_DEFAULT;
    
    if (pI810->pEnt->device->videoRam) {
