--- xc/lib/GL/mesa/drivers/dri/r300/Imakefile.mesa-	2005-11-15 01:09:43.000000000 +0100
+++ xc/lib/GL/mesa/drivers/dri/r300/Imakefile	2005-11-15 01:09:43.000000000 +0100
@@ -0,0 +1,76 @@
+XCOMM $XFree86: xc/lib/GL/mesa/src/drv/r300/Imakefile,v 1.9 3002/02/23 00:45:50 dawes Exp $
+
+#include <Threads.tmpl>
+
+#if GlxUseBuiltInDRIDriver
+#define DoNormalLib (NormalLibGlx || SharedLibGlxWithoutPIC)
+#define DoSharedLib (SharedLibGlx && !SharedLibGlxWithoutPIC)
+#define DoExtraLib SharedLibGlx
+#define DoDebugLib DebugLibGlx
+#define DoProfileLib ProfileLibGlx
+#else
+#define DoNormalLib SharedLibGlxWithoutPIC
+#define DoSharedLib !SharedLibGlxWithoutPIC
+#define DoExtrasLib NO
+#define DoDebugLib NO
+#define DoProfileLib NO
+#endif
+
+#define MesaDrvCommonBuildDir ../common/
+#include "../common/Imakefile.inc"
+#include "../../../array_cache/Imakefile.inc"
+#include "../../../main/Imakefile.inc"
+#include "../../../math/Imakefile.inc"
+#include "../../../shader/Imakefile.inc"
+#include "../../../shader/grammar/Imakefile.inc"
+#include "../../../shader/slang/Imakefile.inc"
+#include "../../../shader/slang/library/Imakefile.inc"
+#include "../../../swrast/Imakefile.inc"
+#include "../../../swrast_setup/Imakefile.inc"
+#include "../../../tnl/Imakefile.inc"
+#include "../../../tnl_dd/Imakefile.inc"
+#include "../../../Imakefile.inc"
+#if defined(i386Architecture) && MesaUseX86Asm
+#include "../../../x86/Imakefile.inc"
+#endif
+#ifdef SparcArchitecture
+#include "../../../sparc/Imakefile.inc"
+#endif
+#define NeedToLinkMesaSrc
+#include "Imakefile.inc"
+
+      DEFINES = $(ALLOC_DEFINES) $(DRI_DEFINES) $(MESA_ASM_DEFINES)
+     INCLUDES = $(X_INCLUDES) $(MESA_INCLUDES) $(DRI_INCLUDES)
+
+      DRMOBJS = $(GLXLIBSRC)/dri/drm/xf86drm.o \
+		$(GLXLIBSRC)/dri/drm/xf86drmHash.o \
+		$(GLXLIBSRC)/dri/drm/xf86drmRandom.o \
+		$(GLXLIBSRC)/dri/drm/xf86drmSL.o
+
+         SRCS = $(R300SRCS)
+         OBJS = $(DRMOBJS) $(COREMESAOBJS) \
+		$(MESA_ASM_OBJS) $(COMMONOBJS) $(R300OBJS) 
+
+REQUIREDLIBS = MathLibrary $(LDPRELIB) $(GLXLIB) $(XONLYLIB) $(EXPATLIB)
+
+#include <Library.tmpl>
+
+LibraryObjectRule()
+
+SubdirLibraryRule($(R300OBJS))
+NormalLintTarget($(SRCS))
+
+#if !GlxUseBuiltInDRIDriver
+LIBNAME = r300_dri.so
+SharedDriModuleTarget($(LIBNAME),DONE $(OBJS),$(OBJS))
+InstallDynamicModule($(LIBNAME),$(MODULEDIR),dri)
+
+#ifdef GlxSoProf
+SOPROF_LIBNAME = _r300_dri_p
+NormalDepLibraryTarget($(SOPROF_LIBNAME),DONE $(OBJS),$(OBJS))
+InstallLibrary($(SOPROF_LIBNAME),$(MODULEDIR)/dri)
+#endif
+#endif
+
+DependTarget()
+
--- xc/lib/GL/mesa/drivers/dri/r300/Imakefile.inc.mesa-	2005-11-15 01:09:43.000000000 +0100
+++ xc/lib/GL/mesa/drivers/dri/r300/Imakefile.inc	2005-11-15 01:42:48.000000000 +0100
@@ -0,0 +1,188 @@
+XCOMM $XFree86: xc/lib/GL/mesa/src/drv/r300/Imakefile.inc,v 1.2 3001/05/02 15:06:04 dawes Exp $
+
+#ifndef MesaDrvSrcDir
+#define MesaDrvSrcDir $(MESASRCDIR)/src/mesa/drivers/dri
+#endif
+
+MESADRVSRCDIR = MesaDrvSrcDir
+
+#ifndef MesaDrvR300BuildDir
+#define MesaDrvR300BuildDir /**/
+#endif
+MESADRVR300BUILDDIR = MesaDrvR300BuildDir
+
+#if Malloc0ReturnsNull
+ALLOC_DEFINES = -DMALLOC_0_RETURNS_NULL
+#endif
+
+#if BuildXF86DRI
+  DRI_DEFINES = GlxDefines -DX_BYTE_ORDER=ByteOrder -DUSE_NEW_INTERFACE -DIN_DRI_DRIVER -DRADEON_COMMON=1 -DRADEON_COMMON_FOR_R300
+ DRI_INCLUDES = -I$(GLXLIBSRC)/dri \
+		-I$(XINCLUDESRC) \
+		-I$(GLXLIBSRC)/glx \
+		-I$(GLXLIBSRC)/include \
+		-I$(SERVERSRC)/GL/dri \
+		-I$(XF86COMSRC) \
+		-I$(XF86OSSRC) \
+		-I$(DRMSRCDIR)/shared-core \
+		-I$(XF86DRIVERSRC)/ati \
+		-I$(GLXLIBSRC)/dri/drm
+#endif
+
+MESA_INCLUDES = -I$(MESASRCDIR)/src/mesa \
+		-I$(MESASRCDIR)/src/mesa/main \
+		-I$(MESASRCDIR)/src/mesa/glapi \
+		-I$(MESASRCDIR)/src/mesa/shader \
+		-I$(MESASRCDIR)/include \
+		-I$(MESADRVSRCDIR)/common \
+		-I$(MESADRVSRCDIR)/r300 \
+		-I$(MESADRVSRCDIR)/radeon \
+                -I$(MESADRVSRCDIR)/radeon/server
+   X_INCLUDES = -I$(XINCLUDESRC) -I$(EXTINCSRC)
+
+   R300SRCS =	$(MESADRVR300BUILDDIR)r300_cmdbuf.c \
+		$(MESADRVR300BUILDDIR)r300_context.c \
+		$(MESADRVR300BUILDDIR)r300_fragprog.c \
+		/* broken? $(MESADRVR300BUILDDIR)r300_fragprog_swz.c */ \
+		$(MESADRVR300BUILDDIR)r300_ioctl.c \
+		$(MESADRVR300BUILDDIR)r300_maos.c \
+		$(MESADRVR300BUILDDIR)r300_render.c \
+		$(MESADRVR300BUILDDIR)r300_shader.c \
+		$(MESADRVR300BUILDDIR)r300_state.c \
+		$(MESADRVR300BUILDDIR)r300_tex.c \
+		$(MESADRVR300BUILDDIR)r300_texmem.c \
+		$(MESADRVR300BUILDDIR)r300_texprog.c \
+		$(MESADRVR300BUILDDIR)r300_texstate.c \
+		$(MESADRVR300BUILDDIR)r300_vertexprog.c \
+		$(MESADRVR300BUILDDIR)radeon_context.c \
+		$(MESADRVR300BUILDDIR)radeon_ioctl.c \
+		$(MESADRVR300BUILDDIR)radeon_lock.c \
+		$(MESADRVR300BUILDDIR)radeon_span.c \
+		$(MESADRVR300BUILDDIR)radeon_screen.c \
+		$(MESADRVR300BUILDDIR)radeon_state.c
+
+   R300OBJS =	$(MESADRVR300BUILDDIR)r300_cmdbuf.o \
+		$(MESADRVR300BUILDDIR)r300_context.o \
+		$(MESADRVR300BUILDDIR)r300_fragprog.o \
+		/* broken? $(MESADRVR300BUILDDIR)r300_fragprog_swz.o */ \
+		$(MESADRVR300BUILDDIR)r300_ioctl.o \
+		$(MESADRVR300BUILDDIR)r300_maos.o \
+		$(MESADRVR300BUILDDIR)r300_render.o \
+		$(MESADRVR300BUILDDIR)r300_shader.o \
+		$(MESADRVR300BUILDDIR)r300_state.o \
+		$(MESADRVR300BUILDDIR)r300_tex.o \
+		$(MESADRVR300BUILDDIR)r300_texmem.o \
+		$(MESADRVR300BUILDDIR)r300_texprog.o \
+		$(MESADRVR300BUILDDIR)r300_texstate.o \
+		$(MESADRVR300BUILDDIR)r300_vertexprog.o \
+		$(MESADRVR300BUILDDIR)radeon_context.o \
+		$(MESADRVR300BUILDDIR)radeon_ioctl.o \
+		$(MESADRVR300BUILDDIR)radeon_lock.o \
+		$(MESADRVR300BUILDDIR)radeon_span.o \
+		$(MESADRVR300BUILDDIR)radeon_screen.o \
+		$(MESADRVR300BUILDDIR)radeon_state.o
+
+   R300UOBJS =	$(MESADRVR300BUILDDIR)unshared/r300_cmdbuf.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_context.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_fragprog.o \
+		/* broken? $(MESADRVR300BUILDDIR)unshared/r300_fragprog_swz.o */ \
+		$(MESADRVR300BUILDDIR)unshared/r300_ioctl.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_maos.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_render.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_shader.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_state.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_tex.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_texmem.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_texprog.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_texstate.o \
+		$(MESADRVR300BUILDDIR)unshared/r300_vertexprog.o \
+		$(MESADRVR300BUILDDIR)unshared/radeon_context.o \
+		$(MESADRVR300BUILDDIR)unshared/radeon_ioctl.o \
+		$(MESADRVR300BUILDDIR)unshared/radeon_lock.o \
+		$(MESADRVR300BUILDDIR)unshared/radeon_span.o \
+		$(MESADRVR300BUILDDIR)unshared/radeon_screen.o \
+		$(MESADRVR300BUILDDIR)unshared/radeon_state.o
+
+   R300DOBJS =	$(MESADRVR300BUILDDIR)debugger/r300_cmdbuf.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_context.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_fragprog.o \
+		/* broken? $(MESADRVR300BUILDDIR)debugger/r300_fragprog_swz.o */ \
+		$(MESADRVR300BUILDDIR)debugger/r300_ioctl.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_maos.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_render.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_shader.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_state.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_tex.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_texmem.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_texprog.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_texstate.o \
+		$(MESADRVR300BUILDDIR)debugger/r300_vertexprog.o \
+		$(MESADRVR300BUILDDIR)debugger/radeon_context.o \
+		$(MESADRVR300BUILDDIR)debugger/radeon_ioctl.o \
+		$(MESADRVR300BUILDDIR)debugger/radeon_lock.o \
+		$(MESADRVR300BUILDDIR)debugger/radeon_span.o \
+		$(MESADRVR300BUILDDIR)debugger/radeon_screen.o \
+		$(MESADRVR300BUILDDIR)debugger/radeon_state.o
+
+   R300POBJS =	$(MESADRVR300BUILDDIR)profiled/r300_cmdbuf.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_context.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_fragprog.o \
+		/* broken? $(MESADRVR300BUILDDIR)profiled/r300_fragprog_swz.o */ \
+		$(MESADRVR300BUILDDIR)profiled/r300_ioctl.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_maos.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_render.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_shader.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_state.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_tex.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_texmem.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_texprog.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_texstate.o \
+		$(MESADRVR300BUILDDIR)profiled/r300_vertexprog.o \
+		$(MESADRVR300BUILDDIR)profiled/radeon_context.o \
+		$(MESADRVR300BUILDDIR)profiled/radeon_ioctl.o \
+		$(MESADRVR300BUILDDIR)profiled/radeon_lock.o \
+		$(MESADRVR300BUILDDIR)profiled/radeon_span.o \
+		$(MESADRVR300BUILDDIR)profiled/radeon_screen.o \
+		$(MESADRVR300BUILDDIR)profiled/radeon_state.o
+
+#ifdef NeedToLinkMesaSrc
+LinkSourceFile(r300_context.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_context.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_cmdbuf.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_cmdbuf.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_emit.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_fixed_pipelines.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_ioctl.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_ioctl.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_maos.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_maos.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_program.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_render.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_reg.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_fragprog.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_fragprog.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_fragprog_swz.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_shader.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_state.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_state.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_tex.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_tex.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_texmem.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_texprog.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_texstate.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(r300_vertexprog.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_context.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_context.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_ioctl.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_ioctl.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_lock.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_lock.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_screen.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_screen.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_span.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_span.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_state.c, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(radeon_state.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(pixel_shader.h, $(MESADRVSRCDIR)/r300)
+LinkSourceFile(vertex_shader.h, $(MESADRVSRCDIR)/r300)
+#endif
