--- xc/config/cf/xorgsite.def~	2005-01-13 23:21:05.000000000 +0100
+++ xc/config/cf/xorgsite.def	2005-01-13 23:29:39.000000000 +0100
@@ -132,8 +132,8 @@
  * work for your architechture.  The old driver will be removed in the
  * next release.
  *
-#define UseDeprecatedKeyboardDriver YES
  */
+#define UseDeprecatedKeyboardDriver YES
 
 /*
  * There are three parameters that determine where and how the Matrox HAL
