--- X11-6.8.1/xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c.orig	2004-10-24 12:16:24.000000000 +0200
+++ X11-6.8.1/xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	2004-10-24 12:17:41.000000000 +0200
@@ -7158,6 +7158,8 @@
     } else
 	if (!RADEONModeInit(pScrn, pScrn->currentMode)) return FALSE;
 
+	RADEONSetFBLocation (pScrn);
+
 #ifdef XF86DRI
     if (info->directRenderingEnabled) {
 	/* get the Radeon back into shape after resume */
