--- xc/config/cf/xorg.tmpl.canonicalize-builderaddr-1-fdo2884-2293	2004-04-23 14:41:59.000000000 -0400
+++ xc/config/cf/xorg.tmpl	2005-04-12 01:20:36.000000000 -0400
@@ -16,7 +16,7 @@
 #ifdef BuilderEMailAddr
 BUILDERADDR = BuilderEMailAddr
 #else 
-BUILDERADDR = "xorg@freedesktop.org"
+BUILDERADDR = "xorg@lists.freedesktop.org"
 #endif
 #ifdef VendorSupportAddress
  VENDORSUPPORT = VendorSupportAddress
