--- xc/programs/xterm/Imakefile.orig	2004-09-13 18:37:31.277489528 +0200
+++ xc/programs/xterm/Imakefile	2004-09-13 18:40:21.011686008 +0200
@@ -18,6 +18,7 @@
 
 #define UseUtempter
 
+COLORS_256 = -DOPT_256_COLORS=1
 /*
  * Fixes to allow compile with X11R5
  */
@@ -124,7 +125,7 @@
     XKB_DEFINES = XkbClientDefines
    PATH_DEFINES = -DPROJECTROOT=$(PROJECTROOT)
         DEFINES = $(XKB_DEFINES) $(TERMCAPDEFINES) $(FEATURE_DEFINES) $(SCROLLBAR_RIGHT) $(UTF8_OPTION) $(XRFDEF) $(PATH_DEFINES) $(TRACEDEF)
-       INCLUDES = -I. -I/usr/include/ncurses $(XRFINCLUDES)
+       INCLUDES = -I. -I/usr/include/ncurses $(XRFINCLUDES) $(COLORS_256)
 
 #ifdef OS2Architecture
         MAINSRC = os2main.c
