diff -ur X11-6.9.0.org/xc/config/cf/Imake.cf X11-6.9.0/xc/config/cf/Imake.cf
--- X11-6.9.0.org/xc/config/cf/Imake.cf	2005-11-08 07:33:24.000000000 +0100
+++ X11-6.9.0/xc/config/cf/Imake.cf	2005-12-22 10:27:11.890963000 +0100
@@ -888,6 +888,10 @@
 #   define s390Architecture
 #  undef __s390__
 # endif /* s390 */
+# ifdef alpha
+#  define AlphaArchitecture
+#  undef alpha
+# endif /* alpha */
 # ifdef __alpha
 #  define AlphaArchitecture
 #  undef __alpha
diff -ur X11-6.9.0.org/xc/config/cf/Imake.tmpl X11-6.9.0/xc/config/cf/Imake.tmpl
--- X11-6.9.0.org/xc/config/cf/Imake.tmpl	2005-11-08 07:33:24.000000000 +0100
+++ X11-6.9.0/xc/config/cf/Imake.tmpl	2005-12-22 10:28:07.490963000 +0100
@@ -266,7 +266,7 @@
 
 #ifndef BuildHtmlManPages
 #ifdef RmanCmd
-#define BuildHtmlManPages YES
+#define BuildHtmlManPages NO
 #else
 #define BuildHtmlManPages NO
 #endif
@@ -882,11 +882,7 @@
 #define LintlibDir $(USRLIBDIR)/lint
 #endif
 #ifndef SystemManDirectory
-#if SystemV4
 #define SystemManDirectory /usr/share/man
-#else
-#define SystemManDirectory /usr/man
-#endif
 #endif
 #ifndef ManDirectoryRoot
 #ifdef ProjectRoot
@@ -1196,7 +1192,7 @@
 #define CppCmd /LibDirName/cpp
 #endif
 #ifndef RawCppCmd
-#define RawCppCmd CppCmd -undef
+#define RawCppCmd CppCmd
 #endif
 #ifndef CppNoLineInfoOption
 #define CppNoLineInfoOption /**/
diff -ur X11-6.9.0.org/xc/config/cf/linux.cf X11-6.9.0/xc/config/cf/linux.cf
--- X11-6.9.0.org/xc/config/cf/linux.cf	2005-10-21 21:10:27.000000000 +0200
+++ X11-6.9.0/xc/config/cf/linux.cf	2005-12-22 10:40:10.100963000 +0100
@@ -394,7 +394,7 @@
 #ifndef BuildLibGlxWithoutPIC
 # if BuildXF86DRI
 #  if defined(i386Architecture)
-#   define BuildLibGlxWithoutPIC	YES
+#   define BuildLibGlxWithoutPIC	NO
 #  endif
 # endif
 #endif
@@ -539,6 +539,10 @@
 # define HasExpat		YES
 #endif
 
+#ifndef HasFontconfig
+# define HasFontconfig         YES
+#endif
+
 #ifndef HasFreetype2
 # define HasFreetype2		YES
 #endif
@@ -792,7 +796,7 @@
 
 #define HasVarRun		YES
 #ifndef VarDbDirectory
-# define VarDbDirectory		$(VARDIR)/lib
+# define VarDbDirectory		/var/lib
 #endif
 
 #ifndef XF86OSCardDrivers
diff -ur X11-6.9.0.org/xc/config/cf/site.def X11-6.9.0/xc/config/cf/site.def
--- X11-6.9.0.org/xc/config/cf/site.def	2004-05-06 17:08:25.000000000 +0200
+++ X11-6.9.0/xc/config/cf/site.def	2005-12-22 10:29:12.390963000 +0100
@@ -124,6 +124,14 @@
 #define DefaultUserPath /bin:/usr/bin:$(BINDIR):/usr/ucb:/usr/local/bin
 */
 
+#define DefaultUserPath /usr/local/bin:/bin:/usr/bin:$(BINDIR)
+#undef DefaultSystemPath
+#define DefaultSystemPath /sbin:/bin:/usr/sbin:/usr/bin:$(BINDIR)
+
+
+#define HasPam    YES
+#define HasPamMisc    YES
+
 
 /* You do NOT need SetUID if you only run the server under xdm */
 /* You MAY need SetUID if users run the server by hand or under xinit */
diff -ur X11-6.9.0.org/xc/config/cf/X11.tmpl X11-6.9.0/xc/config/cf/X11.tmpl
--- X11-6.9.0.org/xc/config/cf/X11.tmpl	2005-12-15 00:41:52.000000000 +0100
+++ X11-6.9.0/xc/config/cf/X11.tmpl	2005-12-22 10:32:25.830963000 +0100
@@ -597,7 +597,7 @@
 #define BuildXF86RushLibrary	(BuildXF86RushExt && !BuildServersOnly)
 #endif
 #ifndef BuildDPSLibraries
-#define BuildDPSLibraries	NO
+#define BuildDPSLibraries	YES
 #endif
 #ifndef BuildDPSLibrary
 #define BuildDPSLibrary		BuildDPSLibraries
@@ -693,7 +693,7 @@
 #endif
 
 #ifndef BuildXterm
-#define BuildXterm		NO
+#define BuildXterm		YES
 #endif
 
 #ifndef BuildXkbcomp
@@ -731,7 +731,7 @@
 #endif
 
 #ifndef InstallHardcopyDocs
-#define InstallHardcopyDocs	YES
+#define InstallHardcopyDocs	NO
 #endif
 
 #ifndef InstallMiscManPages
@@ -1395,8 +1395,12 @@
 # define UsbMouseSupport NO
 #endif
 
+#ifndef EtcX11Directory
+#define EtcX11Directory /etc/X11
+#endif
+
 #ifndef DefaultFSConfigFile
-#define DefaultFSConfigFile $(LIBDIR)/fs/config
+#define DefaultFSConfigFile $(ETCX11DIR)/fs/config
 #endif
 
 #ifndef BinDir
@@ -1483,7 +1487,7 @@
 #define ConfigDir $(LIBDIR)/config
 #endif
 #ifndef DocDir
-#define DocDir $(LIBDIR)/doc
+#define DocDir $(LIBDIR)/share/doc
 #endif
 #ifndef DocHtmlDir
 #define DocHtmlDir $(DOCDIR)/html
@@ -1495,34 +1499,34 @@
 #define DocPdfDir $(DOCDIR)/PDF
 #endif
 #ifndef FontDir
-#define FontDir $(LIBDIR)/fonts
+#define FontDir /usr/share/fonts
 #endif
 #ifndef FontEncDir
 #define FontEncDir $(LIBDIR)/fonts/encodings
 #endif
 #ifndef AdmDir
-#define AdmDir /usr/adm
+#define AdmDir /var/log
 #endif
 #ifndef XinitDir
-#define XinitDir $(LIBDIR)/xinit
+#define XinitDir $(ETCX11DIR)/xinit
 #endif
 #ifndef XdmDir
-#define XdmDir $(LIBDIR)/xdm
+#define XdmDir $(ETCX11DIR)/xdm
 #endif
 #ifndef XdmVarDir
 #define XdmVarDir $(VARLIBDIR)/xdm
 #endif
 #ifndef XdmAuthDir
-#define XdmAuthDir $(LIBDIR)/xdm
+#define XdmAuthDir $(ETCX11DIR)/xdm
 #endif
 #ifndef XdmConfigurationSubdirectory		/* under xdm sources */
 #define XdmConfigurationSubdirectory config
 #endif
 #ifndef TwmDir
-#define TwmDir $(LIBDIR)/twm
+#define TwmDir $(ETCX11DIR)/twm
 #endif
 #ifndef XsmDir
-#define XsmDir $(LIBDIR)/xsm
+#define XsmDir $(ETCX11DIR)/xsm
 #endif
 #ifndef NlsDir
 #define NlsDir $(LIBDIR)/nls
@@ -1502,7 +1502,7 @@
 #define FontDir /usr/share/fonts
 #endif
 #ifndef FontEncDir
-#define FontEncDir $(LIBDIR)/fonts/encodings
+#define FontEncDir /usr/share/fonts/encodings
 #endif
 #ifndef AdmDir
 #define AdmDir /var/log
@@ -1531,23 +1535,20 @@
 #define XLocaleDir $(LIBDIR)/locale
 #endif
 #ifndef LbxproxyDir
-#define LbxproxyDir $(LIBDIR)/lbxproxy
+#define LbxproxyDir $(ETCX11DIR)/lbxproxy
 #endif
 #ifndef LbxproxyConfigSubdir		/* under lbxproxy sources */
 #define LbxproxyConfigSubdir config
 #endif
 #ifndef ProxyManagerDir
-#define ProxyManagerDir $(LIBDIR)/proxymngr
+#define ProxyManagerDir $(ETCX11DIR)/proxymngr
 #endif
 #ifndef ServerConfigDir
-#define ServerConfigDir $(LIBDIR)/xserver
+#define ServerConfigDir $(ETCX11DIR)/xserver
 #endif
 #ifndef XPrintDir
 #define XPrintDir ServerConfigDir
 #endif
-#ifndef EtcX11Directory
-#define EtcX11Directory /etc/X11
-#endif
 #ifndef InstallSecurityConfig
 #define InstallSecurityConfig YES	/* Install sample security policy */
 #endif
@@ -1579,10 +1580,10 @@
 #define OsNameDefines /* as nothing */
 #endif
 #ifndef DefaultUserPath			/* for xdm */
-#define DefaultUserPath :/bin:/usr/bin:$(BINDIR):/usr/ucb
+#define DefaultUserPath /usr/local/bin:/bin:/usr/bin:$(BINDIR)
 #endif
 #ifndef DefaultSystemPath		/* for xdm */
-#define DefaultSystemPath /etc:/bin:/usr/bin:$(BINDIR):/usr/ucb
+#define DefaultSystemPath /sbin:/bin:/usr/sbin:/usr/bin:$(BINDIR)
 #endif
 #ifndef DefaultSystemShell		/* for xdm to set in SHELL */
 #define DefaultSystemShell BourneShell
@@ -1719,7 +1720,7 @@
 
      HTMLINDEXCMD = HtmlIndexCmd
 
-       DOCUTILSRC = $(XTOP)/doc/util
+       DOCUTILSRC = $(XTOP)/share/doc/util
         CLIENTSRC = $(TOP)/clients
           DEMOSRC = $(TOP)/demos
        XDOCMACROS = $(DOCUTILSRC)/macros.t
@@ -1733,7 +1734,7 @@
         SERVERSRC = $(XTOP)/programs/Xserver
        CONTRIBSRC = ContribDir
    UNSUPPORTEDSRC = UnsupportedDir
-           DOCSRC = $(XTOP)/doc
+           DOCSRC = $(XTOP)/share/doc
            RGBSRC = $(XTOP)/programs/rgb
       BDFTOPCFSRC = $(PROGRAMSRC)/bdftopcf
      MKFONTDIRSRC = $(PROGRAMSRC)/mkfontdir
diff -ur X11-6.9.0.org/xc/config/cf/xorg.cf X11-6.9.0/xc/config/cf/xorg.cf
--- X11-6.9.0.org/xc/config/cf/xorg.cf	2005-10-03 18:08:44.000000000 +0200
+++ X11-6.9.0/xc/config/cf/xorg.cf	2005-12-22 10:32:59.590963000 +0100
@@ -136,7 +136,7 @@
 #define XConfigFile		xorg.conf
 #endif
 #ifndef XConfigDir
-#define XConfigDir		$(LIBDIR)
+#define XConfigDir             $(ETCX11DIR)
 #endif
 #ifndef XLogFile
 #define XLogFile		Xorg
@@ -408,7 +408,7 @@
 				neomagic i740 tdfx savage \
 				cirrus vmware tseng trident chips apm \
 				GlideDriver i128 i386Drivers \
-				ati DevelDrivers ark \
+				ati DevelDrivers ark fbdev \
 				cyrix siliconmotion via \
 				vesa vga \
 				dummy XF86OSCardDrivers XF86ExtraCardDrivers
diff -ur X11-6.9.0.org/xc/config/cf/xorgsite.def X11-6.9.0/xc/config/cf/xorgsite.def
--- X11-6.9.0.org/xc/config/cf/xorgsite.def	2005-07-16 09:59:50.000000000 +0200
+++ X11-6.9.0/xc/config/cf/xorgsite.def	2005-12-22 10:33:48.940963000 +0100
@@ -266,8 +266,8 @@
 /*
  * To not build/install the Cyrillic fonts, uncomment the following
  *
-#define BuildCyrillicFonts	NO
  */
+#define BuildCyrillicFonts	YES
 
 /*
  * To not install the local font directory, uncomment the following
@@ -435,8 +435,8 @@
  * Some Linux releases don't have a libtermcap.  In this case you may need
  * to uncomment the following
  *
-#define TermcapLibrary		-lncurses
  */
+#define TermcapLibrary		-lncurses
 
 /*
  * Build a server that dynamically loads the modules by setting
@@ -610,8 +610,8 @@
  * To forceably build static libraries in addition to shared libraries,
  * uncomment this.
  *
-#define ForceNormalLib		YES
  */
+#define ForceNormalLib		YES
 
 /*
  * Uncomment this if your default tools (eg, gcc, ld, as, etc) are
diff -ur X11-6.9.0.org/xc/config/imake/imake.c X11-6.9.0/xc/config/imake/imake.c
--- X11-6.9.0.org/xc/config/imake/imake.c	2005-11-08 07:33:24.000000000 +0100
+++ X11-6.9.0/xc/config/imake/imake.c	2005-12-22 10:34:57.200963000 +0100
@@ -1061,6 +1061,7 @@
   static char* suse = "/etc/SuSE-release";
   static char* redhat = "/etc/redhat-release";
   static char* debian = "/etc/debian_version";
+  static char* pld = "/etc/pld-release";
 
   fprintf (inFile, "%s\n", "#define LinuxUnknown    0");
   fprintf (inFile, "%s\n", "#define LinuxSuSE       1");
@@ -1075,6 +1076,7 @@
   fprintf (inFile, "%s\n", "#define LinuxTurbo      10");
   fprintf (inFile, "%s\n", "#define LinuxWare       11");
   fprintf (inFile, "%s\n", "#define LinuxYggdrasil  12");
+  fprintf (inFile, "%s\n", "#define LinuxPLD        13");
 
 #ifdef CROSSCOMPILE
   if (CrossCompiling) {
@@ -1101,6 +1103,12 @@
      * at the content of /etc/debian_version */
     return;
   }
+  if (lstat (pld, &sb) == 0) {
+    fprintf (inFile, "%s\n", "#define DefaultLinuxDistribution LinuxPLD");
+    fprintf (inFile, "%s\n", "#define DefaultLinuxDistName PLD");
+    return;
+  }
+	  
   /* what's the definitive way to tell what any particular distribution is? */
 
   fprintf (inFile, "%s\n", "#define DefaultLinuxDistribution LinuxUnknown");
diff -ur X11-6.9.0.org/xc/Makefile X11-6.9.0/xc/Makefile
--- X11-6.9.0.org/xc/Makefile	2005-12-15 00:41:52.000000000 +0100
+++ X11-6.9.0/xc/Makefile	2005-12-22 10:39:28.160963000 +0100
@@ -9,7 +9,7 @@
 # or add "MAKE = make" to this file.
 
 RELEASE = "Release 6.9"
-SHELL = /bin/sh
+SHELL = /bin/sh -e
 RM = rm -f
 MV = mv
 WORLDOPTS =
diff -ur X11-6.9.0.org/xc/programs/xdm/config/Xres.cpp X11-6.9.0/xc/programs/xdm/config/Xres.cpp
--- X11-6.9.0.org/xc/programs/xdm/config/Xres.cpp	2004-04-23 21:54:43.000000000 +0200
+++ X11-6.9.0/xc/programs/xdm/config/Xres.cpp	2005-12-22 10:35:32.500963000 +0100
@@ -70,10 +70,10 @@
 #endif /* XPM */
 XHASHendif
 #ifdef XPM
-XHASHif PLANES >= 8
-xlogin*logoFileName: BITMAPDIR/**//XDM_PIXMAP
+XHASHif PLANES > 8
+xlogin*logoFileName: BITMAPDIR/**//xdm-pld-logo.xpm
 XHASHelse
-xlogin*logoFileName: BITMAPDIR/**//XDM_BWPIXMAP
+xlogin*logoFileName: BITMAPDIR/**//xdm-pld-logo-bw.xpm
 XHASHendif
 xlogin*useShape: true
 xlogin*logoPadding: 10
diff -ur X11-6.9.0.org/xc/programs/xdm/resource.c X11-6.9.0/xc/programs/xdm/resource.c
--- X11-6.9.0.org/xc/programs/xdm/resource.c	2005-11-08 07:33:31.000000000 +0100
+++ X11-6.9.0/xc/programs/xdm/resource.c	2005-12-22 10:36:59.580963000 +0100
@@ -85,31 +85,31 @@
  */
 #ifndef __EMX__
 #ifndef DEF_SERVER_LINE 
-#define DEF_SERVER_LINE ":0 local /usr/bin/X11/X :0"
+#define DEF_SERVER_LINE ":0 local /usr/X11R6/bin/X :0"
 #endif
 #ifndef XRDB_PROGRAM
-#define XRDB_PROGRAM "/usr/bin/X11/xrdb"
+#define XRDB_PROGRAM "/usr/X11R6/X11/xrdb"
 #endif
 #ifndef DEF_SESSION
-#define DEF_SESSION "/usr/bin/X11/xterm -ls"
+#define DEF_SESSION "/usr/X11R6/bin/xterm -ls"
 #endif
 #ifndef DEF_USER_PATH
-#define DEF_USER_PATH ":/bin:/usr/bin:/usr/bin/X11:/usr/ucb"
+#define DEF_USER_PATH "/usr/local/bin:/bin:/usr/bin:/usr/X11R6/bin"
 #endif
 #ifndef DEF_SYSTEM_PATH
-#define DEF_SYSTEM_PATH "/etc:/bin:/usr/bin:/usr/bin/X11:/usr/ucb"
+#define DEF_SYSTEM_PATH "/sbin:/bin:/usr/sbin:/usr/bin:/usr/X11R6/bin"
 #endif
 #ifndef DEF_SYSTEM_SHELL
 #define DEF_SYSTEM_SHELL "/bin/sh"
 #endif
 #ifndef DEF_FAILSAFE_CLIENT
-#define DEF_FAILSAFE_CLIENT "/usr/bin/X11/xterm"
+#define DEF_FAILSAFE_CLIENT "/usr/X11R6/bin/xterm"
 #endif
 #ifndef DEF_XDM_CONFIG
-#define DEF_XDM_CONFIG "/usr/lib/X11/xdm/xdm-config"
+#define DEF_XDM_CONFIG "/etc/X11/xdm/xdm-config"
 #endif
 #ifndef DEF_CHOOSER
-#define DEF_CHOOSER "/usr/lib/X11/xdm/chooser"
+#define DEF_CHOOSER "/etc/X11/xdm/chooser"
 #endif
 #ifndef DEF_AUTH_NAME
 #ifdef HASXDMAUTH
@@ -119,7 +119,7 @@
 #endif
 #endif
 #ifndef DEF_AUTH_DIR
-#define DEF_AUTH_DIR "/usr/lib/X11/xdm"
+#define DEF_AUTH_DIR "/etc/X11/xdm"
 #endif
 #ifndef DEF_USER_AUTH_DIR
 #define DEF_USER_AUTH_DIR	"/tmp"
diff -ur X11-6.9.0.org/xc/programs/xinit/Imakefile X11-6.9.0/xc/programs/xinit/Imakefile
--- X11-6.9.0.org/xc/programs/xinit/Imakefile	2005-11-15 05:03:10.000000000 +0100
+++ X11-6.9.0/xc/programs/xinit/Imakefile	2005-12-22 10:37:39.740963000 +0100
@@ -88,11 +88,12 @@
 #endif
 InstallManPage(startx,$(MANDIR))
 
-#if InstallXinitConfig
-InstallNonExecFile($(SAMPLECONFIG),$(XINITDIR))
-#else
-InstallNonExecFileNoClobber($(SAMPLECONFIG),$(XINITDIR))
-#endif
+XCOMM Red Hat gets this from a different package
+XCOMM #if InstallXinitConfig
+XCOMM InstallNonExecFile($(SAMPLECONFIG),$(XINITDIR))
+XCOMM #else
+XCOMM InstallNonExecFileNoClobber($(SAMPLECONFIG),$(XINITDIR))
+XCOMM #endif
 
 LinkConfDirectory(xinit,.,xinit,.)
 
diff -ur X11-6.9.0.org/xc/programs/Xserver/hw/xfree86/xf86cfg/Imakefile X11-6.9.0/xc/programs/Xserver/hw/xfree86/xf86cfg/Imakefile
--- X11-6.9.0.org/xc/programs/Xserver/hw/xfree86/xf86cfg/Imakefile	2005-08-23 03:14:35.000000000 +0200
+++ X11-6.9.0/xc/programs/Xserver/hw/xfree86/xf86cfg/Imakefile	2005-12-22 10:39:11.440963000 +0100
@@ -110,7 +110,7 @@
 
 #if HasNCurses
 CURSESLIB = NCursesLibName
-CURSESDEFINES  = -DHAS_NCURSES
+CURSESDEFINES  = -DHAS_NCURSES -I/usr/include/ncurses
 #endif
 
 XF86CONFIGFILE = XConfigFile
diff -ur X11-6.9.0.org/xc/programs/xterm/Imakefile X11-6.9.0/xc/programs/xterm/Imakefile
--- X11-6.9.0.org/xc/programs/xterm/Imakefile	2005-12-15 00:28:27.000000000 +0100
+++ X11-6.9.0/xc/programs/xterm/Imakefile	2005-12-22 10:38:51.510963000 +0100
@@ -68,6 +68,10 @@
 # endif
 #endif
 
+#if InstallXtermSetUID
+#undef InstallXtermSetUID
+#endif
+
 #ifndef SpecialCObjectRule
 #define SpecialCObjectRule(module,ignore,defines) \
 module.o: ; $(CC) -c defines $(CFLAGS) module.c
@@ -88,6 +92,7 @@
     PUCCPTYDDEF = -DPUCC_PTYD	/* does not need to be setuid */
          PTYLIB = -lpucc
 #endif
+    PTYLIB = -lutil
 
 #if defined(NetBSDArchitecture) || \
     defined(OpenBSDArchitecture) || \
@@ -153,7 +158,7 @@
     XKB_DEFINES = XkbClientDefines
    PATH_DEFINES = -DPROJECTROOT=$(PROJECTROOT)
         DEFINES = $(XKB_DEFINES) $(TERMCAPDEFINES) $(FEATURE_DEFINES) $(SCROLLBAR_RIGHT) $(UTF8_OPTION) $(XRFDEF) $(PATH_DEFINES) $(PUTENVDEF) $(IMAKEDEFINES) $(TRACEDEF)
-       INCLUDES = -I. $(XRFINCLUDES)
+       INCLUDES = -I. -I/usr/include/ncurses $(XRFINCLUDES)
 
 #ifdef OS2Architecture
         MAINSRC = os2main.c
