--- X11-6.8.1/xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c.orig	2004-04-23 21:54:08.000000000 +0200
+++ X11-6.8.1/xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c	2004-10-31 01:40:55.056608568 +0200
@@ -114,6 +114,8 @@
 #endif /* KDKBDREP */
 }
 
+#undef rate
+
 static int
 KIOCSRATE_ioctl_ok(int rate, int delay) {
 #ifdef KIOCSRATE
@@ -141,8 +143,6 @@
 #endif /* KIOCSRATE */
 }
 
-#undef rate
-
 void xf86SetKbdRepeat(char rad)
 {
 #ifdef __sparc__
--- X11-6.8.1/xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c.orig	2004-04-23 21:54:08.000000000 +0200
+++ X11-6.8.1/xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c	2004-10-31 01:41:26.201873768 +0200
@@ -142,6 +142,8 @@
 #endif /* KDKBDREP */
 }
 
+#undef rate
+
 static int
 KIOCSRATE_ioctl_ok(int rate, int delay) {
 #ifdef KIOCSRATE
@@ -169,8 +171,6 @@
 #endif /* KIOCSRATE */
 }
 
-#undef rate
-
 static void
 SetKbdRepeat(InputInfoPtr pInfo, char rad)
 {
