diff -ur X11-6.9.0.org/xc/extras/drm/linux/drm_memory_debug.h X11-6.9.0/xc/extras/drm/linux/drm_memory_debug.h
--- X11-6.9.0.org/xc/extras/drm/linux/drm_memory_debug.h	2005-06-15 20:31:47.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/drm_memory_debug.h	2005-12-22 11:43:11.110963000 +0100
@@ -31,7 +31,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 
 typedef struct drm_mem_stats {
diff -ur X11-6.9.0.org/xc/extras/drm/linux/drm_memory.h X11-6.9.0/xc/extras/drm/linux/drm_memory.h
--- X11-6.9.0.org/xc/extras/drm/linux/drm_memory.h	2005-06-15 20:31:47.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/drm_memory.h	2005-12-22 11:43:11.150963000 +0100
@@ -33,7 +33,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include <linux/highmem.h>
 #include "drmP.h"
 
diff -ur X11-6.9.0.org/xc/extras/drm/linux/drmP.h X11-6.9.0/xc/extras/drm/linux/drmP.h
--- X11-6.9.0.org/xc/extras/drm/linux/drmP.h	2005-10-18 04:49:47.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/drmP.h	2005-12-22 11:43:11.140963000 +0100
@@ -42,7 +42,7 @@
  * can build the DRM (part of PI DRI). 4/21/2000 S + B */
 #include <asm/current.h>
 #endif /* __alpha__ */
-#include <linux/config.h>
+
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/miscdevice.h>
diff -ur X11-6.9.0.org/xc/extras/drm/linux/drm_scatter.h X11-6.9.0/xc/extras/drm/linux/drm_scatter.h
--- X11-6.9.0.org/xc/extras/drm/linux/drm_scatter.h	2005-06-15 20:31:48.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/drm_scatter.h	2005-12-22 11:43:11.130963000 +0100
@@ -31,7 +31,7 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include <linux/vmalloc.h>
 #include "drmP.h"
 
diff -ur X11-6.9.0.org/xc/extras/drm/linux/ffb_drv.c X11-6.9.0/xc/extras/drm/linux/ffb_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/ffb_drv.c	2005-06-15 20:31:49.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/ffb_drv.c	2005-12-22 11:43:11.190963000 +0100
@@ -4,7 +4,7 @@
  * Copyright (C) 2000 David S. Miller (davem@redhat.com)
  */
 
-#include <linux/config.h>
+
 #include "ffb.h"
 #include "drmP.h"
 
diff -ur X11-6.9.0.org/xc/extras/drm/linux/i810_drv.c X11-6.9.0/xc/extras/drm/linux/i810_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/i810_drv.c	2005-06-15 20:31:49.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/i810_drv.c	2005-12-22 11:43:11.120963000 +0100
@@ -30,7 +30,7 @@
  *    Gareth Hughes <gareth@valinux.com>
  */
 
-#include <linux/config.h>
+
 #include "i810.h"
 #include "drmP.h"
 #include "drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux/i830_drv.c X11-6.9.0/xc/extras/drm/linux/i830_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/i830_drv.c	2005-06-15 20:31:50.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/i830_drv.c	2005-12-22 11:43:11.150963000 +0100
@@ -32,7 +32,7 @@
  *    Keith Whitwell <keith@tungstengraphics.com>
  */
 
-#include <linux/config.h>
+
 #include "i830.h"
 #include "drmP.h"
 #include "drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux/mach64_drv.c X11-6.9.0/xc/extras/drm/linux/mach64_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/mach64_drv.c	2005-06-15 20:31:47.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/mach64_drv.c	2005-12-22 11:43:11.170963000 +0100
@@ -27,7 +27,7 @@
  *    Leif Delgass <ldelgass@retinalburn.net>
  */
 
-#include <linux/config.h>
+
 #include "mach64.h"
 #include "drmP.h"
 #include "drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux/mga_drv.c X11-6.9.0/xc/extras/drm/linux/mga_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/mga_drv.c	2005-06-15 20:31:47.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/mga_drv.c	2005-12-22 11:43:11.210963000 +0100
@@ -29,7 +29,7 @@
  *    Gareth Hughes <gareth@valinux.com>
  */
 
-#include <linux/config.h>
+
 #include "mga.h"
 #include "drmP.h"
 #include "drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux/r128_drv.c X11-6.9.0/xc/extras/drm/linux/r128_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/r128_drv.c	2005-06-15 20:31:49.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/r128_drv.c	2005-12-22 11:43:11.170963000 +0100
@@ -29,7 +29,7 @@
  *    Gareth Hughes <gareth@valinux.com>
  */
 
-#include <linux/config.h>
+
 #include "r128.h"
 #include "drmP.h"
 #include "drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux/radeon_drv.c X11-6.9.0/xc/extras/drm/linux/radeon_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/radeon_drv.c	2005-06-15 20:31:48.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/radeon_drv.c	2005-12-22 11:43:11.210963000 +0100
@@ -30,7 +30,7 @@
  */
 
 
-#include <linux/config.h>
+
 #include "radeon.h"
 #include "drmP.h"
 #include "drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux/savage_drv.c X11-6.9.0/xc/extras/drm/linux/savage_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/savage_drv.c	2005-06-15 20:31:49.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/savage_drv.c	2005-12-22 11:43:11.160963000 +0100
@@ -22,7 +22,7 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include "savage.h"
 #include "drmP.h"
 #include "savage_drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux/sis_drv.c X11-6.9.0/xc/extras/drm/linux/sis_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/sis_drv.c	2005-06-15 20:31:48.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/sis_drv.c	2005-12-22 11:43:11.240963000 +0100
@@ -25,7 +25,7 @@
  *
  */
 
-#include <linux/config.h>
+
 #include "sis.h"
 #include "drmP.h"
 #include "sis_drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux/tdfx_drv.c X11-6.9.0/xc/extras/drm/linux/tdfx_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux/tdfx_drv.c	2005-06-15 20:31:47.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux/tdfx_drv.c	2005-12-22 11:43:11.220963000 +0100
@@ -30,7 +30,7 @@
  *    Gareth Hughes <gareth@valinux.com>
  */
 
-#include <linux/config.h>
+
 #include "tdfx.h"
 #include "drmP.h"
 
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/drm_memory.c X11-6.9.0/xc/extras/drm/linux-core/drm_memory.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/drm_memory.c	2005-11-30 23:59:11.000000000 +0100
+++ X11-6.9.0/xc/extras/drm/linux-core/drm_memory.c	2005-12-22 11:43:10.970963000 +0100
@@ -33,7 +33,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include <linux/highmem.h>
 #include "drmP.h"
 
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/drm_memory_debug.c X11-6.9.0/xc/extras/drm/linux-core/drm_memory_debug.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/drm_memory_debug.c	2005-11-30 23:59:11.000000000 +0100
+++ X11-6.9.0/xc/extras/drm/linux-core/drm_memory_debug.c	2005-12-22 11:43:10.840963000 +0100
@@ -31,7 +31,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 
 #ifdef DEBUG_MEMORY
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/drm_memory_debug.h X11-6.9.0/xc/extras/drm/linux-core/drm_memory_debug.h
--- X11-6.9.0.org/xc/extras/drm/linux-core/drm_memory_debug.h	2005-10-18 04:24:08.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/drm_memory_debug.h	2005-12-22 11:43:10.890963000 +0100
@@ -31,7 +31,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 
 typedef struct drm_mem_stats {
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/drm_memory.h X11-6.9.0/xc/extras/drm/linux-core/drm_memory.h
--- X11-6.9.0.org/xc/extras/drm/linux-core/drm_memory.h	2005-10-18 04:24:08.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/drm_memory.h	2005-12-22 11:43:10.930963000 +0100
@@ -33,7 +33,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include <linux/highmem.h>
 #include <linux/vmalloc.h>
 #include "drmP.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/drmP.h X11-6.9.0/xc/extras/drm/linux-core/drmP.h
--- X11-6.9.0.org/xc/extras/drm/linux-core/drmP.h	2005-11-30 23:59:10.000000000 +0100
+++ X11-6.9.0/xc/extras/drm/linux-core/drmP.h	2005-12-22 11:43:10.930963000 +0100
@@ -41,7 +41,7 @@
  * can build the DRM (part of PI DRI). 4/21/2000 S + B */
 #include <asm/current.h>
 #endif				/* __alpha__ */
-#include <linux/config.h>
+
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/miscdevice.h>
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/drm_scatter.c X11-6.9.0/xc/extras/drm/linux-core/drm_scatter.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/drm_scatter.c	2005-10-18 04:24:09.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/drm_scatter.c	2005-12-22 11:43:11.000963000 +0100
@@ -31,7 +31,7 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include <linux/vmalloc.h>
 
 #include "drmP.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/drm_sysfs.c X11-6.9.0/xc/extras/drm/linux-core/drm_sysfs.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/drm_sysfs.c	2005-10-18 04:24:09.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/drm_sysfs.c	2005-12-22 11:43:10.870963000 +0100
@@ -11,7 +11,7 @@
  *
  */
 
-#include <linux/config.h>
+
 #include <linux/device.h>
 #include <linux/kdev_t.h>
 #include <linux/err.h>
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/ffb_drv.c X11-6.9.0/xc/extras/drm/linux-core/ffb_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/ffb_drv.c	2005-10-18 04:24:09.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/ffb_drv.c	2005-12-22 11:43:10.980963000 +0100
@@ -4,7 +4,7 @@
  * Copyright (C) 2000 David S. Miller (davem@redhat.com)
  */
 
-#include <linux/config.h>
+
 #include <linux/sched.h>
 #include <linux/smp_lock.h>
 #include <asm/shmparam.h>
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/i810_drv.c X11-6.9.0/xc/extras/drm/linux-core/i810_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/i810_drv.c	2005-10-18 04:24:09.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/i810_drv.c	2005-12-22 11:43:10.900963000 +0100
@@ -30,7 +30,7 @@
  *    Gareth Hughes <gareth@valinux.com>
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "drm.h"
 #include "i810_drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/i830_drv.c X11-6.9.0/xc/extras/drm/linux-core/i830_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/i830_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/i830_drv.c	2005-12-22 11:43:10.950963000 +0100
@@ -32,7 +32,7 @@
  *    Keith Whitwell <keith@tungstengraphics.com>
  */
 
-#include <linux/config.h>
+
 
 #include "drmP.h"
 #include "drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/imagine_drv.c X11-6.9.0/xc/extras/drm/linux-core/imagine_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/imagine_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/imagine_drv.c	2005-12-22 11:43:10.960963000 +0100
@@ -22,7 +22,7 @@
 
 /* derived from tdfx_drv.c */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "imagine_drv.h"
 
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/mach64_drv.c X11-6.9.0/xc/extras/drm/linux-core/mach64_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/mach64_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/mach64_drv.c	2005-12-22 11:43:10.970963000 +0100
@@ -27,7 +27,7 @@
  *    Leif Delgass <ldelgass@retinalburn.net>
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "drm.h"
 #include "mach64_drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/mga_drv.c X11-6.9.0/xc/extras/drm/linux-core/mga_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/mga_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/mga_drv.c	2005-12-22 11:43:11.000963000 +0100
@@ -29,7 +29,7 @@
  *    Gareth Hughes <gareth@valinux.com>
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "drm.h"
 #include "mga_drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/nv_drv.c X11-6.9.0/xc/extras/drm/linux-core/nv_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/nv_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/nv_drv.c	2005-12-22 11:43:10.910963000 +0100
@@ -32,7 +32,7 @@
  *    Lars Knoll <lars@trolltech.com>
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "nv_drv.h"
 
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/r128_drv.c X11-6.9.0/xc/extras/drm/linux-core/r128_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/r128_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/r128_drv.c	2005-12-22 11:43:10.960963000 +0100
@@ -29,7 +29,7 @@
  *    Gareth Hughes <gareth@valinux.com>
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "drm.h"
 #include "r128_drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/radeon_drv.c X11-6.9.0/xc/extras/drm/linux-core/radeon_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/radeon_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/radeon_drv.c	2005-12-22 11:43:10.980963000 +0100
@@ -29,7 +29,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "drm.h"
 #include "radeon_drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/savage_drv.c X11-6.9.0/xc/extras/drm/linux-core/savage_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/savage_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/savage_drv.c	2005-12-22 11:43:10.950963000 +0100
@@ -23,7 +23,7 @@
  * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "savage_drm.h"
 #include "savage_drv.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/sis_drv.c X11-6.9.0/xc/extras/drm/linux-core/sis_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/sis_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/sis_drv.c	2005-12-22 11:43:11.020963000 +0100
@@ -25,7 +25,7 @@
  *
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "sis_drm.h"
 #include "sis_drv.h"
diff -ur X11-6.9.0.org/xc/extras/drm/linux-core/tdfx_drv.c X11-6.9.0/xc/extras/drm/linux-core/tdfx_drv.c
--- X11-6.9.0.org/xc/extras/drm/linux-core/tdfx_drv.c	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/linux-core/tdfx_drv.c	2005-12-22 11:43:11.010963000 +0100
@@ -30,7 +30,7 @@
  *    Gareth Hughes <gareth@valinux.com>
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "tdfx_drv.h"
 
diff -ur X11-6.9.0.org/xc/extras/drm/shared/drm.h X11-6.9.0/xc/extras/drm/shared/drm.h
--- X11-6.9.0.org/xc/extras/drm/shared/drm.h	2005-07-27 02:15:46.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/shared/drm.h	2005-12-22 11:43:11.400963000 +0100
@@ -64,7 +64,7 @@
 
 #if defined(__linux__)
 #if defined(__KERNEL__)
-#include <linux/config.h>
+
 #endif
 #include <asm/ioctl.h>		/* For _IO* macros */
 #define DRM_IOCTL_NR(n)		_IOC_NR(n)
diff -ur X11-6.9.0.org/xc/extras/drm/shared/via_drv.c X11-6.9.0/xc/extras/drm/shared/via_drv.c
--- X11-6.9.0.org/xc/extras/drm/shared/via_drv.c	2005-06-15 20:31:59.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/shared/via_drv.c	2005-12-22 11:43:11.350963000 +0100
@@ -22,7 +22,7 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include "via.h"
 #include "drmP.h"
 #include "via_drm.h"
diff -ur X11-6.9.0.org/xc/extras/drm/shared-core/drm.h X11-6.9.0/xc/extras/drm/shared-core/drm.h
--- X11-6.9.0.org/xc/extras/drm/shared-core/drm.h	2005-10-18 04:24:10.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/shared-core/drm.h	2005-12-22 11:43:10.740963000 +0100
@@ -70,7 +70,7 @@
 
 #if defined(__linux__)
 #if defined(__KERNEL__)
-#include <linux/config.h>
+
 #endif
 #include <asm/ioctl.h>		/* For _IO* macros */
 #define DRM_IOCTL_NR(n)		_IOC_NR(n)
diff -ur X11-6.9.0.org/xc/extras/drm/shared-core/via_drv.c X11-6.9.0/xc/extras/drm/shared-core/via_drv.c
--- X11-6.9.0.org/xc/extras/drm/shared-core/via_drv.c	2005-10-18 04:24:15.000000000 +0200
+++ X11-6.9.0/xc/extras/drm/shared-core/via_drv.c	2005-12-22 11:43:10.710963000 +0100
@@ -22,7 +22,7 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-#include <linux/config.h>
+
 #include "drmP.h"
 #include "via_drm.h"
 #include "via_drv.h"
diff -ur X11-6.9.0.org/xc/programs/Xserver/hw/xfree86/drivers/sis/init301.h X11-6.9.0/xc/programs/Xserver/hw/xfree86/drivers/sis/init301.h
--- X11-6.9.0.org/xc/programs/Xserver/hw/xfree86/drivers/sis/init301.h	2005-09-18 01:07:17.000000000 +0200
+++ X11-6.9.0/xc/programs/Xserver/hw/xfree86/drivers/sis/init301.h	2005-12-22 11:43:11.410963000 +0100
@@ -67,7 +67,7 @@
 #ifdef SIS_CP
 #undef SIS_CP
 #endif
-#include <linux/config.h>
+
 #include <linux/version.h>
 #include <linux/types.h>
 #include <asm/io.h>
diff -ur X11-6.9.0.org/xc/programs/Xserver/hw/xfree86/drivers/sis/init.h X11-6.9.0/xc/programs/Xserver/hw/xfree86/drivers/sis/init.h
--- X11-6.9.0.org/xc/programs/Xserver/hw/xfree86/drivers/sis/init.h	2005-09-07 21:44:14.000000000 +0200
+++ X11-6.9.0/xc/programs/Xserver/hw/xfree86/drivers/sis/init.h	2005-12-22 11:43:11.440963000 +0100
@@ -73,7 +73,7 @@
 #ifdef SIS_CP
 #undef SIS_CP
 #endif
-#include <linux/config.h>
+
 #include <linux/version.h>
 #include <linux/types.h>
 #include <asm/io.h>
diff -ur X11-6.9.0.org/xc/programs/Xserver/hw/xfree86/drivers/sis/osdef.h X11-6.9.0/xc/programs/Xserver/hw/xfree86/drivers/sis/osdef.h
--- X11-6.9.0.org/xc/programs/Xserver/hw/xfree86/drivers/sis/osdef.h	2005-07-04 12:57:08.000000000 +0200
+++ X11-6.9.0/xc/programs/Xserver/hw/xfree86/drivers/sis/osdef.h	2005-12-22 11:43:11.450963000 +0100
@@ -90,7 +90,7 @@
 /**********************************************************************/
 
 #ifdef SIS_LINUX_KERNEL
-#include <linux/config.h>
+
 #include <linux/version.h>
 
 #ifdef CONFIG_FB_SIS_300
