

http://lists.freedesktop.org/archives/xorg/2005-April/007250.html

Index: xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c
===================================================================
--- xc.orig/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	2005-04-04 10:18:39.000000000 +1000
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	2005-04-04 10:35:07.000000000 +1000
@@ -1006,6 +1006,9 @@
 	MonType = MT_NONE;
     }
 
+    OUTREG(info->DDCReg, INREG(info->DDCReg) &
+	   ~(RADEON_GPIO_EN_0 | RADEON_GPIO_EN_1));
+
     if (*MonInfo) {
 	if ((*MonInfo)->rawData[0x14] & 0x80) {
 	    /* Note some laptops have a DVI output that uses internal TMDS,
