diff -ur X11-6.8.1/xc/include/Xmd.h X11-modified/xc/include/Xmd.h
--- X11-6.8.1/xc/include/Xmd.h	2005-02-05 16:37:53 +0000
+++ X11-modified/xc/include/Xmd.h	2005-02-05 16:13:30 +0000
@@ -58,6 +58,11 @@
 #ifdef CRAY
 #define WORD64				/* 64-bit architecture */
 #endif
+
+#if defined(__sparc_v9__)
+#define __sparc64__
+#endif
+
 #if defined (_LP64) || \
     defined(__alpha) || defined(__alpha__) || \
     defined(__ia64__) || defined(ia64) || \
