--- xc.orig/programs/xman/Imakefile	2004-08-08 14:22:59.000000000 +0200
+++ xc/programs/xman/Imakefile	2004-08-08 15:01:51.815244864 +0200
@@ -26,7 +26,7 @@
 # if defined(LinuxDistribution) && (LinuxDistribution == LinuxRedHat)
     MANCONF = -DMANCONF=\"/etc/man.config\"
 # else
-    MANCONF = -DMANCONF=\"/etc/man.conf\"
+    MANCONF = -DMANCONF=\"/etc/man.config\"
 # endif
 #elif defined(OpenBSDArchitecture) || defined(NetBSDArchitecture) || \
     defined(DarwinArchitecture)
--- xc.orig/programs/xman/xman.man	2004-08-08 14:22:59.000000000 +0200
+++ xc/programs/xman/xman.man	2004-08-08 15:01:51.820244104 +0200
@@ -93,8 +93,8 @@
 .SH "BSD AND LINUX SYSTEMS"        
 .PP
 In newer BSD and Linux systems, \fIXman\fP will search for a file named
-\fI/etc/man.conf\fP which will contain the list of directories containing
-manual pages. See \fIman.conf\fP(5) for a complete description of the file
+\fI/etc/man.config\fP which will contain the list of directories containing
+manual pages. See \fIman.config\fP(5) for a complete description of the file
 format.
 .SH "THE MANDESC FILE"        
 .PP
