diff -aurN xc.orig/lib/Xft/xftfreetype.c xc/lib/Xft/xftfreetype.c
--- xc.orig/lib/Xft/xftfreetype.c	2004-08-12 01:37:34.000000000 +0200
+++ xc/lib/Xft/xftfreetype.c	2005-03-28 23:05:32.300707656 +0200
@@ -290,7 +290,7 @@
 	    FT_Done_Face (f->face);
     }
     XftMemFree (XFT_MEM_FILE,
-		sizeof (XftFtFile) + f->file ? strlen (f->file) + 1 : 0);
+		sizeof (XftFtFile) + (f->file ? strlen (f->file) + 1 : 0));
     free (f);
 }
 
