--- xc/Makefile.makefile-fastbuild	2003-02-26 04:21:31.000000000 -0500
+++ xc/Makefile	2003-04-06 11:40:20.000000000 -0400
@@ -83,9 +83,9 @@
 	$(MAKE) $(MFLAGS) Makefile.boot
 	$(MAKE_CMD) $(MFLAGS) VerifyOS
 	$(MAKE_CMD) $(MFLAGS) Makefiles
-	$(MAKE_CMD) $(MFLAGS) BOOTSTRAPSUBDIRS= clean
+	if [ -z "$(FAST)" ]; then $(MAKE_CMD) $(MFLAGS) BOOTSTRAPSUBDIRS= clean; fi
 	$(MAKE_CMD) $(MFLAGS) includes
-	$(MAKE_CMD) $(MFLAGS) depend
+	if [ -z "$(FAST)" ]; then $(MAKE_CMD) $(MFLAGS) depend; fi
 	$(MAKE_CMD) $(MFLAGS) $(WORLDOPTS) World
 	@echo ""
 	@date
