--- xc/programs/Xserver/hw/xfree86/loader/dixsym.c.orig	2004-08-08 21:14:05.762968728 +0200
+++ xc/programs/Xserver/hw/xfree86/loader/dixsym.c	2004-08-08 21:10:42.751831112 +0200
@@ -81,6 +81,7 @@
 #include "dgaproc.h"
 #ifdef RENDER
 #include "mipict.h"
+#include "renderedge.h"
 #endif
 #include "selection.h"
 #ifdef XKB
@@ -400,6 +401,10 @@
     SYMFUNC(PictureSetFilterAlias)
     SYMFUNC(PictureGetSubpixelOrder)
     SYMFUNC(PictureSetSubpixelOrder)
+    SYMFUNC(RenderLineFixedEdgeInit)
+    SYMFUNC(RenderSampleFloorY)
+    SYMFUNC(RenderSampleCeilY)
+    SYMFUNC(RenderEdgeInit)
 #endif
 
     {0, 0}
