--- xc/config/cf/xorg.cf~	2004-08-08 18:31:40.144527928 +0200
+++ xc/config/cf/xorg.cf	2004-08-08 18:31:40.160525496 +0200
@@ -1145,25 +1145,25 @@
  * By default, build all of the fonts.
  */
 #ifndef Build75DpiFonts
-# define Build75DpiFonts	YES
+# define Build75DpiFonts	NO
 #endif
 #ifndef Build100DpiFonts
-# define Build100DpiFonts	YES
+# define Build100DpiFonts	NO
 #endif
 #ifndef BuildSpeedoFonts
 # define BuildSpeedoFonts	NO
 #endif
 #ifndef BuildType1Fonts
-# define BuildType1Fonts	YES
+# define BuildType1Fonts	NO
 #endif
 #ifndef BuildCIDFonts
-# define BuildCIDFonts		YES
+# define BuildCIDFonts		NO
 #endif
 #ifndef BuildTrueTypeFonts
-# define BuildTrueTypeFonts	YES
+# define BuildTrueTypeFonts	NO
 #endif
 #ifndef BuildCyrillicFonts
-# define BuildCyrillicFonts	YES
+# define BuildCyrillicFonts	NO
 #endif
 
 /*
