--- X11-6.9.0/xc/programs/xkbcomp/symbols/level3.old	2005-08-12 13:10:11.000000000 +0200
+++ X11-6.9.0/xc/programs/xkbcomp/symbols/level3	2006-01-31 20:05:00.000000000 +0100
@@ -12,7 +12,7 @@
 default partial modifier_keys
 xkb_symbols "ralt_switch" {
   key <RALT> {
-    type[Group1]="ONE_LEVEL",
+    type[Group1]="TWO_LEVEL",
     symbols[Group1] = [ ISO_Level3_Shift ]
     };
   modifier_map Mod5   { ISO_Level3_Shift };
