--- xc/programs/xterm/Imakefile.orig	2004-09-13 18:37:31.277489528 +0200
+++ xc/programs/xterm/Imakefile	2004-09-13 18:40:21.011686008 +0200
@@ -25,6 +25,9 @@
         UTMPLIB = -lutempter
 #endif
 
+#define UseUtempter
+
+COLORS_256 = -DOPT_256_COLORS=1
 /*
  * setgid mode works for systems that do not require setuid to open pty.
  */
@@ -158,7 +161,7 @@
     XKB_DEFINES = XkbClientDefines
    PATH_DEFINES = -DPROJECTROOT=$(PROJECTROOT)
         DEFINES = $(XKB_DEFINES) $(TERMCAPDEFINES) $(FEATURE_DEFINES) $(SCROLLBAR_RIGHT) $(UTF8_OPTION) $(XRFDEF) $(PATH_DEFINES) $(PUTENVDEF) $(IMAKEDEFINES) $(TRACEDEF)
-       INCLUDES = -I. -I/usr/include/ncurses $(XRFINCLUDES)
+       INCLUDES = -I. -I/usr/include/ncurses $(XRFINCLUDES) $(COLORS_256)
 
 #ifdef OS2Architecture
         MAINSRC = os2main.c

