--- XFree86-4.1.0/xc/config/cf/xf86site.def.orig	Tue Jul  3 08:44:25 2001
+++ XFree86-4.1.0/xc/config/cf/xf86site.def	Tue Jul  3 08:44:36 2001
@@ -251,7 +251,7 @@
  * will be included in it.  When building the loadable server each of these
  * modules will be built.
  *
-#define XF86CardDrivers		mga glint nv tga s3virge sis rendition \
+#define XF86CardDrivers		mga glint nv tga s3 s3virge sis rendition \
 				neomagic i740 tdfx savage \
 				cirrus vmware tseng trident chips apm \
 				GlideDriver fbdev i128 \
--- XFree86-4.1.0/xc/config/cf/xfree86.cf.orig	Tue Jul  3 10:10:40 2001
+++ XFree86-4.1.0/xc/config/cf/xfree86.cf	Tue Jul  3 10:10:42 2001
@@ -302,7 +302,7 @@
 
 /* Pure PCI drivers should go first */
 # ifndef XF86CardDrivers
-#  define XF86CardDrivers	mga glint nv tga s3virge sis rendition \
+#  define XF86CardDrivers	mga glint nv tga s3 s3virge sis rendition \
 				neomagic i740 tdfx savage \
 				cirrus vmware tseng trident chips apm \
 				GlideDriver fbdev i128 \
@@ -472,7 +472,7 @@
 # endif
 
 # ifndef XF86CardDrivers
-#  define XF86CardDrivers	mga glint nv s3virge sis savage \
+#  define XF86CardDrivers	mga glint nv s3 s3virge sis savage \
 				trident chips tdfx fbdev ati \
 				DevelDrivers newport \
 				XF86OSCardDrivers XF86ExtraCardDrivers
@@ -555,7 +555,7 @@
 
 /* Pure PCI drivers should go first */
 # ifndef XF86CardDrivers
-#  define XF86CardDrivers	mga tdfx glint s3virge rendition tga \
+#  define XF86CardDrivers	mga tdfx glint s3 s3virge rendition tga \
 				savage nv DevelDrivers siliconmotion vga \
 				XF86OSCardDrivers XF86ExtraCardDrivers
 # endif
@@ -648,7 +648,7 @@
 
 /* Pure PCI drivers should go first */
 # ifndef XF86CardDrivers
-#  define XF86CardDrivers	mga glint s3virge sis savage\
+#  define XF86CardDrivers	mga glint s3 s3virge sis savage\
 				trident chips tdfx fbdev \
 				DevelDrivers vga \
 				XF86OSCardDrivers XF86ExtraCardDrivers
