diff -Nur xc/programs/xfs.org/difs/main.c xc/programs/xfs/difs/main.c
--- xc/programs/xfs.org/difs/main.c	Fri Apr  7 10:14:45 2000
+++ xc/programs/xfs/difs/main.c	Fri Apr  7 10:02:01 2000
@@ -56,6 +56,9 @@
 #include	"dispatch.h"
 #include	"extentst.h"
 #include	"difs.h"
+
+#include	"debug.h"
+
 #include <unistd.h>
 #include <pwd.h>
 #include <sys/types.h>
@@ -77,6 +80,7 @@
 static Bool create_connection_block(void);
 
 char       *configfilename;
+int	   debug_level; 
 extern Bool drone_server;
 
 extern OldListenRec *OldListen;
@@ -92,6 +96,7 @@
     argvGlobal = argv;
     droppriv = 0;
     becomeDaemon = 0;
+    debug_level = 0;
 
     configfilename = DEFAULT_CONFIG_FILE;
 
diff -Nur xc/programs/xfs.org/include/debug.h xc/programs/xfs/include/debug.h
--- xc/programs/xfs.org/include/debug.h	Thu Jan  1 01:00:00 1970
+++ xc/programs/xfs/include/debug.h	Fri Apr  7 10:02:35 2000
@@ -0,0 +1 @@
+/* debug.h */
\ No newline at end of file
diff -Nur xc/programs/xfs.org/os/utils.c xc/programs/xfs/os/utils.c
--- xc/programs/xfs.org/os/utils.c	Fri Apr  7 10:14:45 2000
+++ xc/programs/xfs/os/utils.c	Fri Apr  7 10:04:38 2000
@@ -91,6 +91,7 @@
 #endif
 
 extern char *configfilename;
+extern int  debug_level;
 extern int  droppriv; /* whether or not to drop root privileges */
 extern int  becomeDaemon; /* whether to become a daemon or not */
 char       *progname;
@@ -191,7 +192,7 @@
 static void
 usage(void)
 {
-    fprintf(stderr, "usage: %s [-config config_file] [-port tcp_port] [-droppriv] [-daemon]\n",
+    fprintf(stderr, "usage: %s [-config config_file] [-port tcp_port] [-droppriv] [-daemon] [-d debug_level]\n",
 	    progname);
     exit(1);
 }
@@ -295,6 +296,8 @@
 		ProcessLSoption (argv[++i]);
 	    else
 		usage();
+	} else if (!strcmp(argv[i], "-d") || !strcmp(argv[i], "--debug")) {
+		debug_level = atoi(argv[++i]);
 	} else if (!strcmp(argv[i], "-droppriv")) {
 	        droppriv = 1;
 	} else if (!strcmp(argv[i], "-daemon")) {
