--- xc/lib/Xcursor/Imakefile.orig	2005-10-25 18:58:48.000000000 +0200
+++ xc/lib/Xcursor/Imakefile	2006-06-11 03:46:50.029912096 +0200
@@ -72,7 +72,7 @@
 	RemoveFile($@)
 	sh config-subst $(SUBSTVARS) < xcursor.pc.in > $@
 
-InstallNonExecFile(xcursor.pc,$(PKGCONFIGDIR))
+InstallNonExecFile(xcursor.pc,/usr/lib/pkgconfig)
 
 clean::
 	RemoveFile(xcursor.pc)
--- xc/lib/Xft/Imakefile.orig	2005-10-25 18:58:48.000000000 +0200
+++ xc/lib/Xft/Imakefile	2006-06-11 03:47:20.687251472 +0200
@@ -135,7 +135,7 @@
 	RemoveFile($@)
 	sh config/config-subst $(SUBSTVARS) < xft.pc.in > $@
 
-InstallNonExecFile(xft.pc,$(PKGCONFIGDIR))
+InstallNonExecFile(xft.pc,/usr/lib/pkgconfig)
 
 clean::
 	RemoveFile(xft.pc)
