--- XFree86-4.3.0/xc/programs/xman/vendor.h.orig	Sat Sep 21 04:38:25 2002
+++ XFree86-4.3.0/xc/programs/xman/vendor.h	Wed May 14 23:04:25 2003
@@ -71,7 +71,7 @@
 #  define SYSMANPATH "/usr/catman/u_man:/usr/catman/a_man"
 #elif defined(__bsdi__)
 #  define SYSMANPATH "/usr/share/man:/usr/contrib/man:/usr/contrib/isode/man:/usr/local/man"
-#elif defined(__OpenBSD__) || defined(__DARWIN__)
+#elif defined(__OpenBSD__) || defined(__DARWIN__) || defined(linux)
 #  define SYSMANPATH "/usr/share/man:/usr/local/man:/usr/X11R6/man"
 #elif defined(SVR4) || defined(__osf__) || (defined(BSD) && (BSD >= 199103))
 #  define SYSMANPATH "/usr/share/man"
