--- XFree86-4.2.99.901/xc/lib/GL/mesa/src/Imakefile.inc.orig	Thu Nov 14 22:01:16 2002
+++ XFree86-4.2.99.901/xc/lib/GL/mesa/src/Imakefile.inc	Thu Feb 13 21:02:28 2003
@@ -76,10 +76,6 @@
 COREMESASRCS = $(COREMESABASESRCS) \
 		$(MESABUILDDIR)array_cache/ac_context.c \
 		$(MESABUILDDIR)array_cache/ac_import.c \
-		$(MESABUILDDIR)math/m_debug_clip.c \
-		$(MESABUILDDIR)math/m_debug_norm.c \
-		$(MESABUILDDIR)math/m_debug_vertex.c \
-		$(MESABUILDDIR)math/m_debug_xform.c \
 		$(MESABUILDDIR)math/m_eval.c \
 		$(MESABUILDDIR)math/m_matrix.c \
 		$(MESABUILDDIR)math/m_translate.c \
--- XFree86-4.2.99.901/xc/lib/GL/mesa/src/math/Imakefile.inc.orig	Thu Nov 14 22:01:17 2002
+++ XFree86-4.2.99.901/xc/lib/GL/mesa/src/math/Imakefile.inc	Fri Feb 14 08:01:51 2003
@@ -30,10 +30,7 @@
 LinkSourceFile(m_xform.c, $(MESASRCDIR)/src/math)
 #endif
 
-  MESA_MATH_OBJS = $(MESAMATHBUILDDIR)m_debug_clip.o \
-		   $(MESAMATHBUILDDIR)m_debug_norm.o \
-		   $(MESAMATHBUILDDIR)m_debug_xform.o \
-		   $(MESAMATHBUILDDIR)m_debug_vertex.o \
+  MESA_MATH_OBJS = \
 		   $(MESAMATHBUILDDIR)m_eval.o \
 		   $(MESAMATHBUILDDIR)m_matrix.o \
 		   $(MESAMATHBUILDDIR)m_translate.o \
@@ -42,10 +39,7 @@
 		   $(MESAMATHBUILDDIR)m_xform.o
 
 #if defined(DoSharedLib) && DoSharedLib
- MESA_MATH_UOBJS = $(MESAMATHBUILDDIR)unshared/m_debug_clip.o \
-		   $(MESAMATHBUILDDIR)unshared/m_debug_norm.o \
-		   $(MESAMATHBUILDDIR)unshared/m_debug_xform.o \
-		   $(MESAMATHBUILDDIR)unshared/m_debug_vertex.o \
+ MESA_MATH_UOBJS = \
 		   $(MESAMATHBUILDDIR)unshared/m_eval.o \
 		   $(MESAMATHBUILDDIR)unshared/m_matrix.o \
 		   $(MESAMATHBUILDDIR)unshared/m_translate.o \
