--- XFree86-4.1.0.orig/xc/programs/xman/xman.man	Sat Jan 27 19:21:18 2001
+++ XFree86-4.1.0/xc/programs/xman/xman.man	Sun Jun 17 22:41:25 2001
@@ -93,8 +93,8 @@
 .SH "BSD AND LINUX SYSTEMS"        
 .PP
 In newer BSD and Linux systems, \fIXman\fP will search for a file named
-\fI/etc/man.conf\fP which will contain the list of directories containing
-manual pages. See \fIman.conf\fP(5) for a complete description of the file
+\fI/etc/man.config\fP which will contain the list of directories containing
+manual pages. See \fIman.config\fP(5) for a complete description of the file
 format.
 .SH "THE MANDESC FILE"        
 .PP
