--- xc/programs/xman/vendor.h	2006-01-09 16:01:38.000000000 +0100
+++ xc/programs/xman/vendor.h-new	2006-06-11 03:25:03.629515144 +0200
@@ -70,7 +70,7 @@
 #  define SYSMANPATH "/usr/catman/u_man:/usr/catman/a_man"
 #elif defined(__bsdi__)
 #  define SYSMANPATH "/usr/share/man:/usr/contrib/man:/usr/contrib/isode/man:/usr/local/man"
-#elif defined(__OpenBSD__) || defined(__DARWIN__)
+#elif defined(__OpenBSD__) || defined(__DARWIN__) || defined(linux)
 #  define SYSMANPATH "/usr/share/man:/usr/local/man:/usr/X11R6/man"
 #elif defined(SVR4) || defined(__osf__) || (defined(BSD) && (BSD >= 199103))
 #  define SYSMANPATH "/usr/share/man"
