--- XFree86-4.2.1/xc/lib/font/Type1/t1funcs.c.orig	Fri Dec 14 20:56:44 2001
+++ XFree86-4.2.1/xc/lib/font/Type1/t1funcs.c	Wed Sep 18 16:04:20 2002
@@ -523,7 +523,9 @@
        if (hypot(vals->pixel_matrix[0], vals->pixel_matrix[1]) < 1.0 ||
 	   hypot(vals->pixel_matrix[2], vals->pixel_matrix[3]) < 1.0)
 	   return BadFontName;
-
+       /* Reject large sizes too - may cause DoS in xfs or X-server due to abort() */
+       if (hypot(vals->pixel_matrix[0], vals->pixel_matrix[3]) > 5000.0)
+	   return BadFontName;
        /* set up default values */
        FontDefaultFormat(&bit, &byte, &glyph, &scan);
        /* get any changes made from above */
