--- XFree86-4.0/xc/config/cf/X11.tmpl	Wed Jun  7 07:29:47 2000
+++ XFree86-4.0/xc/config/cf/X11.tmpl	Wed Jun  7 07:35:17 2000
@@ -1016,7 +1016,7 @@
 #endif
 #ifndef XFileSearchPathBase
 /* avoid SCCS keyword */
-#define XFileSearchPathBase Concat4($(LIBDIR)/%L/%T/%N%C,%S:$(LIBDIR)/%l/%T/%N%C,%S:$(LIBDIR)/%T/%N%C,%S:$(LIBDIR)/%L/%T/%N%S:$(LIBDIR)/%l/%T/%N%S:$(LIBDIR)/%T/%N%S)
+#define XFileSearchPathBase $(LIBDIR)/%T/%L/%N%C%S:$(LIBDIR)/%T/%l/%N%C%S:$(LIBDIR)/%T/C/%N%C%S:$(LIBDIR)/%T/POSIX/%N%C%S:$(LIBDIR)/%T/%N%C%S:$(LIBDIR)/%T/%L/%N%S:$(LIBDIR)/%T/%l/%N%S:$(LIBDIR)/%T/C/%N%S:$(LIBDIR)/%T/POSIX/%N%S:$(LIBDIR)/%T/%N%S:$(LIBDIR)/%L/%T/%N%C%S:$(LIBDIR)/%l/%T/%N%C%S:$(LIBDIR)/%L/%T/%N%S:$(LIBDIR)/%l/%T/%N%S
 #endif
 #ifndef XFileSearchPathDefault
 #ifdef XAppLoadDir
@@ -2199,11 +2199,9 @@
 #ifndef InstallAppDefaultsLong
 #if InstallAppDefFiles
 #define InstallAppDefaultsLong(file,class)				@@\
-LinkConfDirectory(app-defaults,.,app-defaults,.)			@@\
 InstallNamedTarget(install,file.ad,$(INSTAPPFLAGS),$(XAPPLOADDIR),class)
 #else
 #define InstallAppDefaultsLong(file,class)				@@\
-LinkConfDirectory(app-defaults,.,app-defaults,.)			@@\
 InstallNamedTargetNoClobber(install,file.ad,$(INSTAPPFLAGS),$(XAPPLOADDIR),class)
 #endif /* InstallAppDefFiles */
 #endif /* InstallAppDefaultsLong */
