--- XFree86-4.2.1/xc/lib/GL/mesa/src/Imakefile.inc.orig	Tue Apr  3 00:15:47 2001
+++ XFree86-4.2.1/xc/lib/GL/mesa/src/Imakefile.inc	Wed Sep 18 16:11:39 2002
@@ -197,7 +197,6 @@
 		$(MESABUILDDIR)context.o \
 		$(MESABUILDDIR)copypix.o \
 		$(MESABUILDDIR)cva.o \
-		$(MESABUILDDIR)debug_xform.o \
 		$(MESABUILDDIR)depth.o \
 		$(MESABUILDDIR)dlist.o \
 		$(DISPATCHOBJS) \
@@ -282,7 +281,6 @@
 		$(MESABUILDDIR)unshared/context.o \
 		$(MESABUILDDIR)unshared/copypix.o \
 		$(MESABUILDDIR)unshared/cva.o \
-		$(MESABUILDDIR)unshared/debug_xform.o \
 		$(MESABUILDDIR)unshared/depth.o \
 		$(DISPATCHUOBJS) \
 		$(MESABUILDDIR)unshared/dlist.o \
