diff -urN xc.orig/programs/xdm/Imakefile xc/programs/xdm/Imakefile
--- xc.orig/programs/xdm/Imakefile	Mon Dec 30 15:54:06 2002
+++ xc/programs/xdm/Imakefile	Mon Dec 30 18:33:35 2002
@@ -256,9 +256,9 @@
 		  '-DDEF_SYSTEM_SHELL="$(BOURNE_SHELL)"' \
 		  '-DDEF_FAILSAFE_CLIENT="$(BINDIR)/xterm"' \
 		  '-DDEF_XDM_CONFIG="$(XDMDIR)/xdm-config"' \
-		  '-DDEF_CHOOSER="$(CHOOSERPATH)"' \
-		  '-DDEF_AUTH_DIR="$(XDMDIR)"' \
-		  '-DDEF_GREETER_LIB="$(XDMDIR)/libXdmGreet.so"'
+		  '-DDEF_CHOOSER="$(BINDIR)/chooser"' \
+		  '-DDEF_AUTH_DIR="$(XDMVARDIR)"' \
+		  '-DDEF_GREETER_LIB="$(USRLIBDIR)/libXdmGreet.so"'
 
 ComplexProgramTarget_1(xdm,$(LOCAL_LIBRARIES),NullParameter)
 LinkConfDirectory(xdm,.,xdm,.)
@@ -277,7 +277,7 @@
 #if BuildChooser
 SpecialCObjectRule(chooser, $(ICONFIGFILES), $(SOCK_DEFINES))
 NormalProgramTarget(chooser,$(OBJS3),$(DEPLIBS3),$(LIBS3),NullParameter)
-InstallProgram(chooser,$(XDMDIR))
+InstallProgram(chooser,$(BINDIR))
 InstallAppDefaults(Chooser)
 #endif
 #if BuildBoth
diff -urN xc.org/programs/xdm/config/Xres.cpp xc/programs/xdm/config/Xres.cpp
--- xc.org/programs/xdm/config/Xres.cpp	2003-11-23 23:32:35.000000000 +0100
+++ xc/programs/xdm/config/Xres.cpp	2003-11-23 23:33:41.000000000 +0100
@@ -41,9 +41,7 @@
 XHASHendif
 #endif /* XPM */
 XHASHifdef COLOR
-#ifndef XPM
-xlogin*greetColor: CadetBlue
-#else
+#ifdef XPM
 xlogin*borderWidth: 1
 xlogin*frameWidth: 5
 xlogin*innerFramesWidth: 2
@@ -52,6 +50,8 @@
 xlogin*background: grey
 !xlogin*foreground: darkgreen
 xlogin*greetColor: Blue3
+#else
+login*greetColor: CadetBlue
 #endif /* XPM */
 xlogin*failColor: red
 *Foreground: black
--- xc/programs/xdm/config/Imakefile.orig	2006-01-09 16:01:05.000000000 +0100
+++ xc/programs/xdm/config/Imakefile	2006-06-11 02:49:16.806881792 +0200
@@ -4,13 +4,7 @@
 
 all:: Xservers.ws xdm-config Xservers Xresources
 
-#if (defined(i386Architecture) || defined(AMD64Architecture)) && (defined(KNetBSDArchitecture) || defined(KOpenBSDArchitecture))
-DEFAULTVT=vt05
-#endif
-
-#if defined(LinuxDistribution) && (LinuxDistribution == LinuxDebian)
-DEFAULTVT=vt7
-#endif
+DEFAULTVT=vt9
 
 #ifndef XdmPixmap
 # define XdmPixmap XFree86.xpm
@@ -36,7 +30,11 @@
 #endif
 
 CppFileTarget(Xservers.ws,Xserv.ws.cpp,-DBINDIR=$(BINDIR) -DDEFAULTVT=$(DEFAULTVT),NullParameter)
-CppFileTarget(xdm-config,xdm-conf.cpp,-DXDMDIR=$(XDMDIR) -DXDMAUTHDIR=$(XDMDIR) -DXDMLOGDIR=$(XDMLOGDIR) -DXDMPIDDIR=$(XDMPIDDIR) -DSU=$(SU) -DCHOOSERPATH=$(CHOOSERPATH),NullParameter)
+#ifdef LinuxArchitecture
+CppFileTarget(xdm-config,xdm-conf.cpp,$(XPM_DEFINES) -DLINUX=1 -DXDMDIR=$(XDMDIR) -DXDMAUTHDIR=$(XDMDIR) -DXDMLOGDIR=$(XDMLOGDIR) -DXDMPIDDIR=$(XDMPIDDIR) -DSU=$(SU) -DXDMVARDIR=$(XDMVARDIR) -DCHOOSERPATH="$(BINDIR)/chooser",NullParameter)
+#else
+CppFileTarget(xdm-config,xdm-conf.cpp,$(XPM_DEFINES) -DXDMDIR=$(XDMDIR) -DXDMAUTHDIR=$(XDMDIR) -DXDMLOGDIR=$(XDMLOGDIR) -DXDMPIDDIR=$(XDMPIDDIR) -DSU=$(SU) -DCHOOSERPATH="$(BINDIR)/chooser",NullParameter)
+#endif /* LinuxArchitecture */
 CppFileTarget(Xresources,Xres.cpp,$(XPM_DEFINES),NullParameter)
 
 LinkFile(Xservers,Xservers.$(SERVERSTYPE))
