--- XFree86-4.3.99.14/xc/lib/GL/mesa/src/math/Imakefile.inc.orig	2003-10-10 13:02:44.000000000 +0200
+++ XFree86-4.3.99.14/xc/lib/GL/mesa/src/math/Imakefile.inc	2003-10-12 14:33:09.000000000 +0200
@@ -26,20 +23,14 @@
 LinkSourceFile(m_xform.c, $(MESASRCDIR)/src/math)
 #endif
 
-  MESA_MATH_OBJS = $(MESAMATHBUILDDIR)m_debug_clip.o \
-		   $(MESAMATHBUILDDIR)m_debug_norm.o \
-		   $(MESAMATHBUILDDIR)m_debug_xform.o \
-		   $(MESAMATHBUILDDIR)m_eval.o \
+  MESA_MATH_OBJS = $(MESAMATHBUILDDIR)m_eval.o \
 		   $(MESAMATHBUILDDIR)m_matrix.o \
 		   $(MESAMATHBUILDDIR)m_translate.o \
 		   $(MESAMATHBUILDDIR)m_vector.o \
 		   $(MESAMATHBUILDDIR)m_xform.o
 
 #if defined(DoSharedLib) && DoSharedLib
- MESA_MATH_UOBJS = $(MESAMATHBUILDDIR)unshared/m_debug_clip.o \
-		   $(MESAMATHBUILDDIR)unshared/m_debug_norm.o \
-		   $(MESAMATHBUILDDIR)unshared/m_debug_xform.o \
-		   $(MESAMATHBUILDDIR)unshared/m_eval.o \
+ MESA_MATH_UOBJS = $(MESAMATHBUILDDIR)unshared/m_eval.o \
 		   $(MESAMATHBUILDDIR)unshared/m_matrix.o \
 		   $(MESAMATHBUILDDIR)unshared/m_translate.o \
 		   $(MESAMATHBUILDDIR)unshared/m_vector.o \
@@ -48,19 +39,13 @@
  MATH_MESA_UOBJS = $(MESA_MATH_OBJS)
 #endif
 
- MESA_MATH_DOBJS = $(MESAMATHBUILDDIR)debugger/m_debug_clip.o \
-		   $(MESAMATHBUILDDIR)debugger/m_debug_norm.o \
-		   $(MESAMATHBUILDDIR)debugger/m_debug_xform.o \
-		   $(MESAMATHBUILDDIR)debugger/m_eval.o \
+ MESA_MATH_DOBJS = $(MESAMATHBUILDDIR)debugger/m_eval.o \
 		   $(MESAMATHBUILDDIR)debugger/m_matrix.o \
 		   $(MESAMATHBUILDDIR)debugger/m_translate.o \
 		   $(MESAMATHBUILDDIR)debugger/m_vector.o \
 		   $(MESAMATHBUILDDIR)debugger/m_xform.o
 
- MESA_MATH_POBJS = $(MESAMATHBUILDDIR)profiled/m_debug_clip.o \
-		   $(MESAMATHBUILDDIR)profiled/m_debug_norm.o \
-		   $(MESAMATHBUILDDIR)profiled/m_debug_xform.o \
-		   $(MESAMATHBUILDDIR)profiled/m_eval.o \
+ MESA_MATH_POBJS = $(MESAMATHBUILDDIR)profiled/m_eval.o \
 		   $(MESAMATHBUILDDIR)profiled/m_matrix.o \
 		   $(MESAMATHBUILDDIR)profiled/m_translate.o \
 		   $(MESAMATHBUILDDIR)profiled/m_vector.o \
