--- XFree86-4.3.0/xc/programs/xfs.org/difs/main.c	Fri Apr  7 10:14:45 2000
+++ XFree86-4.3.0/xc/programs/xfs/difs/main.c	Fri Apr  7 10:02:01 2000
@@ -56,6 +56,7 @@
 #include	"dispatch.h"
 #include	"extentst.h"
 #include	"difs.h"
+#include	"debug.h"
 
 char       *ConnectionInfo;
 int         ConnInfoLen;
@@ -77,6 +80,7 @@
 static Bool create_connection_block(void);
 
 char       *configfilename;
+int	   debug_level; 
 extern Bool drone_server;
 
 extern OldListenRec *OldListen;
@@ -92,6 +96,7 @@
     int         i, oldumask;
 
     argcGlobal = argc;
+    debug_level = 0;
     argvGlobal = argv;
 
     configfilename = DEFAULT_CONFIG_FILE;
diff -Nur xc/programs/xfs.org/include/debug.h xc/programs/xfs/include/debug.h
--- xc/programs/xfs.org/include/debug.h	Thu Jan  1 01:00:00 1970
+++ xc/programs/xfs/include/debug.h	Fri Apr  7 10:02:35 2000
@@ -0,0 +1 @@
+/* debug.h */
\ No newline at end of file
--- XFree86-4.3.0/xc/programs/xfs/os./utils.c.orig	Tue Oct 15 03:45:03 2002
+++ XFree86-4.3.0/xc/programs/xfs/os/utils.c	Wed May 14 17:12:58 2003
@@ -93,6 +93,7 @@
 #include <stdlib.h>
 
 extern char *configfilename;
+extern int  debug_level;
 static Bool dropPriv = FALSE; /* whether or not to drop root privileges */
 #ifdef DEFAULT_DAEMON
 static Bool becomeDaemon = TRUE; /* whether to become a daemon or not */
@@ -219,7 +219,7 @@
 static void
 usage(void)
 {
-    fprintf(stderr, "usage: %s [-config config_file] [-port tcp_port] [-droppriv] [-daemon] [-nodaemon] [-user user_name] [-ls listen_socket]\n",
+    fprintf(stderr, "usage: %s [-config config_file] [-port tcp_port] [-droppriv] [-daemon] [-nodaemon] [-user user_name] [-ls listen_socket] [-d debug_level]\n",
 	    progname);
     exit(1);
 }
@@ -333,6 +335,8 @@
 		ProcessLSoption (argv[++i]);
 	    else
 		usage();
+	} else if (!strcmp(argv[i], "-d") || !strcmp(argv[i], "--debug")) {
+		debug_level = atoi(argv[++i]);
 	} else if (!strcmp(argv[i], "-droppriv")) {
 	        dropPriv = TRUE;
 	} else if (!strcmp(argv[i], "-daemon")) {
