Index: mga_dri.c
===================================================================
RCS file: /home/x-cvs/xc/programs/Xserver/hw/xfree86/drivers/mga/mga_dri.c,v
retrieving revision 1.14
diff -u -r1.14 mga_dri.c
--- XFree86-4.0.2/xc/programs/Xserver/hw/xfree86/drivers/mga/mga_dri.c	2000/12/11 22:34:55	1.14
+++ XFree86-4.0.2/xc/programs/Xserver/hw/xfree86/drivers/mga/mga_dri.c	2001/03/02 20:13:06
@@ -379,6 +379,7 @@
    int init_offset;
    int i;
    unsigned long mode_mask;
+   CARD32 pciCommand;
 
    switch(pMGA->Chipset) {
    case PCI_CHIP_MGAG400:
@@ -782,6 +783,11 @@
       return FALSE;
    }
    xf86DrvMsg(pScrn->scrnIndex, X_INFO, "visual configs initialized\n" );
+
+   /* Enable bus mastering in PCI config space */
+   pciCommand = pciReadLong(pMGA->PciTag, PCI_CMD_STAT_REG);
+   pciWriteLong(pMGA->PciTag, PCI_CMD_STAT_REG,
+		pciCommand | PCI_CMD_MASTER_ENABLE);
 
    return TRUE;
 }
