--- XFree86-4.0.1/xc/programs/xman/vendor.h.foo	Mon Aug 14 15:13:05 2000
+++ XFree86-4.0.1/xc/programs/xman/vendor.h	Mon Aug 14 15:13:38 2000
@@ -60,7 +60,7 @@
 #endif
 
 #if (defined(BSD) && (BSD >= 199103)) || defined(linux)
-# define MANCONF "/etc/man.conf"
+# define MANCONF "/etc/man.config"
 #endif
 
 /*
--- XFree86-4.0.1/xc/programs/xman/xman.man.foo	Mon Aug 14 15:13:11 2000
+++ XFree86-4.0.1/xc/programs/xman/xman.man	Mon Aug 14 15:13:24 2000
@@ -90,8 +90,8 @@
 .SH "BSD AND LINUX SYSTEMS"        
 .PP
 In newer BSD and Linux systems, \fIXman\fP will search for a file named
-\fI/etc/man.conf\fP which will contain the list of directories containing
-manual pages. See \fIman.conf\fP(5) for a complete description of the file
+\fI/etc/man.config\fP which will contain the list of directories containing
+manual pages. See \fIman.config\fP(5) for a complete description of the file
 format.
 .SH "THE MANDESC FILE"        
 .PP
