--- XFree86-4.3.0/xc/config/cf/Library.tmpl.orig	Sat Jan  4 03:48:24 2003
+++ XFree86-4.3.0/xc/config/cf/Library.tmpl	Wed May 14 22:28:11 2003
@@ -212,9 +212,9 @@
 #  if !DoSharedLib &&  defined(IncludeSharedObjectInNormalLib)
 #   define _NormalObjCompile(options) NormalRelocLibObjCompile(options)
 #  else
-#   define _NormalObjCompile(options) NormalLibObjCompile(options)
+#   define _NormalObjCompile(options) NormalLibObjCompile(options $(PICFLAGS))
 #  endif
-#  define _NormalObjCplusplusCompile(options) NormalLibObjCplusplusCompile(options)
+#  define _NormalObjCplusplusCompile(options) NormalLibObjCplusplusCompile(options $(PICFLAGS))
 #  define _NormalCleanDir() $(_NULLCMD_)
 # endif
 #endif
