--- xc/lib/Xi/XListDev.c.orig	2003-04-30 02:40:02.000000000 -0400
+++ xc/lib/Xi/XListDev.c	2003-04-30 02:40:27.000000000 -0400
@@ -81,10 +81,10 @@
     register long 		rlen;
     XExtDisplayInfo *info = XInput_find_display (dpy);
 
-    LockDisplay (dpy);
     if (_XiCheckExtInit(dpy, XInput_Initial_Release) == -1)
 	return ((XDeviceInfo *) NULL);
 
+    LockDisplay (dpy);
     GetReq(ListInputDevices,req);		
     req->reqType = info->codes->major_opcode;
     req->ReqType = X_ListInputDevices;
