--- XFree86-4.0/xc/lib/dps/Xlibnet.h.fhs	Thu May 18 19:46:12 2000
+++ XFree86-4.0/xc/lib/dps/Xlibnet.h	Wed Jun 28 19:17:19 2000
@@ -20,7 +20,7 @@
 
 #ifndef X_UNIX_PATH
 #ifdef hpux
-#define X_UNIX_PATH "/usr/spool/sockets/X11/"
+#define X_UNIX_PATH "/var/spool/sockets/X11/"
 #define OLD_UNIX_PATH "/tmp/.X11-unix/X"
 #else
 #define X_UNIX_PATH "/tmp/.X11-unix/X"
--- XFree86-4.0/xc/lib/dps/csconndi.c.fhs	Thu May 18 19:46:12 2000
+++ XFree86-4.0/xc/lib/dps/csconndi.c	Wed Jun 28 19:17:19 2000
@@ -483,8 +483,8 @@
 
 #ifndef CSDPS_UNIX_PATH
 #ifdef hpux
-#define CSDPS_UNIX_DIR      "/usr/spool/sockets/DPSNX"
-#define CSDPS_UNIX_PATH     "/usr/spool/sockets/DPSNX/"
+#define CSDPS_UNIX_DIR      "/var/spool/sockets/DPSNX"
+#define CSDPS_UNIX_PATH     "/var/spool/sockets/DPSNX/"
 #else
 #define CSDPS_UNIX_DIR      "/tmp/.DPSNX-unix"
 #define CSDPS_UNIX_PATH     "/tmp/.DPSNX-unix/AGENT"
--- XFree86-4.0/xc/lib/xtrans/Xtranssock.c.fhs	Thu May 11 14:14:10 2000
+++ XFree86-4.0/xc/lib/xtrans/Xtranssock.c	Wed Jun 28 19:17:19 2000
@@ -203,30 +203,30 @@
 #ifdef hpux
 
 #if defined(X11_t)
-#define UNIX_PATH "/usr/spool/sockets/X11/"
-#define UNIX_DIR "/usr/spool/sockets/X11"
+#define UNIX_PATH "/var/spool/sockets/X11/"
+#define UNIX_DIR "/var/spool/sockets/X11"
 #define OLD_UNIX_PATH "/tmp/.X11-unix/X"
 #endif /* X11_t */
 #if defined(XIM_t)
-#define UNIX_PATH "/usr/spool/sockets/XIM/"
-#define UNIX_DIR "/usr/spool/sockets/XIM"
+#define UNIX_PATH "/var/spool/sockets/XIM/"
+#define UNIX_DIR "/var/spool/sockets/XIM"
 #define OLD_UNIX_PATH "/tmp/.XIM-unix/XIM"
 #endif /* XIM_t */
 #if defined(FS_t) || defined(FONT_t)
-#define UNIX_PATH "/usr/spool/sockets/fontserv/"
-#define UNIX_DIR "/usr/spool/sockets/fontserv"
+#define UNIX_PATH "/var/spool/sockets/fontserv/"
+#define UNIX_DIR "/var/spool/sockets/fontserv"
 #endif /* FS_t || FONT_t */
 #if defined(ICE_t)
-#define UNIX_PATH "/usr/spool/sockets/ICE/"
-#define UNIX_DIR "/usr/spool/sockets/ICE"
+#define UNIX_PATH "/var/spool/sockets/ICE/"
+#define UNIX_DIR "/var/spool/sockets/ICE"
 #endif /* ICE_t */
 #if defined(TEST_t)
-#define UNIX_PATH "/usr/spool/sockets/xtrans_test/"
-#define UNIX_DIR "/usr/spool/sockets/xtrans_test"
+#define UNIX_PATH "/var/spool/sockets/xtrans_test/"
+#define UNIX_DIR "/var/spool/sockets/xtrans_test"
 #endif
 #if defined(LBXPROXY_t)
-#define UNIX_PATH "/usr/spool/sockets/X11/"
-#define UNIX_DIR  "/usr/spool/sockets/X11"
+#define UNIX_PATH "/var/spool/sockets/X11/"
+#define UNIX_DIR  "/var/spool/sockets/X11"
 #endif
 
 #else /* !hpux */
--- XFree86-4.0/xc/programs/rstart/Imakefile.fhs	Wed Jun 28 19:34:12 2000
+++ XFree86-4.0/xc/programs/rstart/Imakefile	Wed Jun 28 19:34:32 2000
@@ -39,7 +39,7 @@
 CLIENTNAME=rstart.cmd
 #endif
 
-MYLIBDIR=$(LIBDIR)/rstart
+MYLIBDIR=/etc/X11/rstart
 ENVPREFIX=RSTART
 PACKAGEname=rstart
 
--- XFree86-4.0/xc/programs/xbiff/xbiff.man.fhs	Sat Feb 12 22:56:13 2000
+++ XFree86-4.0/xc/programs/xbiff/xbiff.man	Wed Jun 28 19:17:19 2000
@@ -55,7 +55,7 @@
 .TP 8
 .B \-file \fIfilename\fP
 This option specifies the name of the file which should be monitored.  By
-default, it watches /usr/spool/mail/\fIusername\fP, where \fIusername\fP
+default, it watches /var/mail/\fIusername\fP, where \fIusername\fP
 is your login name.
 .TP 8
 .B \-volume \fIpercentage\fP
@@ -114,7 +114,7 @@
 .TP 8
 .B file (\fPclass\fB File)
 Specifies the name of the file to monitor.  The default is to watch
-/usr/spool/mail/\fIusername\fP, where \fIusername\fP is your login name.
+/var/mail/\fIusername\fP, where \fIusername\fP is your login name.
 .TP 8
 .B onceOnly (\fPclass\fB Boolean)
 Specifies that the bell is only rung the first time new mail is found
--- XFree86-4.0/xc/programs/xbiff/MailboxP.h.fhs	Sat Feb 12 22:56:12 2000
+++ XFree86-4.0/xc/programs/xbiff/MailboxP.h	Wed Jun 28 19:17:19 2000
@@ -50,7 +50,7 @@
 #endif
 
 #ifndef MAILBOX_DIRECTORY
-#define MAILBOX_DIRECTORY "/usr/spool/mail"
+#define MAILBOX_DIRECTORY "/var/mail"
 #endif
 
 typedef struct {			/* new fields for mailbox widget */
--- XFree86-4.0/xc/programs/xmh/xmh.man.fhs	Sat Dec 21 22:31:12 1996
+++ XFree86-4.0/xc/programs/xmh/xmh.man	Wed Jun 28 19:17:19 2000
@@ -1313,7 +1313,7 @@
 An example of \fI.xmhcheck\fP file format, for the folders ``inbox'' and
 ``xpert'':
 .nf
-inbox	/usr/spool/mail/converse
+inbox	/var/mail/converse
 xpert	/users/converse/maildrops/xpert
 .fi
 .sp
