diff -urN xc.orig/programs/xman/Imakefile xc/programs/xman/Imakefile
--- xc.orig/programs/xman/Imakefile	Mon Dec 30 15:54:14 2002
+++ xc/programs/xman/Imakefile	Mon Dec 30 18:21:38 2002
@@ -26,7 +26,7 @@
 # if defined(LinuxDistribution) && (LinuxDistribution == LinuxRedHat)
     MANCONF = -DMANCONF=\"/etc/man.config\"
 # else
-    MANCONF = -DMANCONF=\"/etc/man.conf\"
+    MANCONF = -DMANCONF=\"/etc/man.config\"
 # endif
 #elif defined(OpenBSDArchitecture) || defined(NetBSDArchitecture) || \
     defined(DarwinArchitecture)
