--- XFree86-4.1.0/xc/config/cf/xf86site.def.orig	Tue Jul  3 08:44:25 2001
+++ XFree86-4.1.0/xc/config/cf/xf86site.def	Tue Jul  3 08:44:36 2001
@@ -251,7 +251,7 @@
  * will be included in it.  When building the loadable server each of these
  * modules will be built.
  *
-#define XF86CardDrivers		mga glint nv tga s3virge sis rendition \
+#define XF86CardDrivers		mga glint nv tga s3 s3virge sis rendition \
 				neomagic i740 tdfx savage \
 				cirrus vmware tseng trident chips apm \
 				GlideDriver fbdev i128 \
