diff -durN -x '*~' XFree86-4.3.99.901.orig/xc/programs/Xserver/hw/xfree86/os-support/shared/drm/kernel/drm.h XFree86-4.3.99.901/xc/programs/Xserver/hw/xfree86/os-support/shared/drm/kernel/drm.h
--- XFree86-4.3.99.901.orig/xc/programs/Xserver/hw/xfree86/os-support/shared/drm/kernel/drm.h	2003-12-02 14:02:43.000000000 +0100
+++ XFree86-4.3.99.901/xc/programs/Xserver/hw/xfree86/os-support/shared/drm/kernel/drm.h	2004-12-20 16:14:32.996352304 +0100
@@ -38,7 +38,6 @@
 #define _DRM_H_
 
 #if defined(__linux__)
-#include <linux/config.h>
 #include <asm/ioctl.h>		/* For _IO* macros */
 #define DRM_IOCTL_NR(n)		_IOC_NR(n)
 #define DRM_IOC_VOID		_IOC_NONE
