diff -Nru xc.orig/config/cf/xfree86.cf xc/config/cf/xfree86.cf
--- xc.orig/config/cf/xfree86.cf	2003-09-25 13:00:04.000000000 +0200
+++ xc/config/cf/xfree86.cf	2003-09-26 10:57:00.626294344 +0200
@@ -374,7 +374,7 @@
 #  define XF86CardDrivers	mga glint nv tga s3 s3virge sis rendition \
 				neomagic i740 tdfx savage \
 				cirrus vmware tseng trident chips apm \
-				GlideDriver i128 NscDriver \
+				i128 NscDriver \
 				ati i810 AgpGartDrivers DevelDrivers ark \
 				cyrix siliconmotion via \
 				fbdev vesa vga \
