--- XFree86-4.3.99.902/xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c.orig	2003-11-21 06:59:50.000000000 +0100
+++ XFree86-4.3.99.902/xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c	2004-01-18 10:49:38.166058944 +0100
@@ -115,6 +115,8 @@
 #endif /* KDKBDREP */
 }
 
+#undef rate
+
 static int
 KIOCSRATE_ioctl_ok(int rate, int delay) {
 #ifdef KIOCSRATE
@@ -141,8 +143,6 @@
 #endif /* KIOCSRATE */
 }
 
-#undef rate
-
 void xf86SetKbdRepeat(char rad)
 {
 #ifdef __sparc__
--- XFree86-4.3.99.902/xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c.orig	2003-11-21 06:59:50.000000000 +0100
+++ XFree86-4.3.99.902/xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c	2004-02-06 22:32:27.511210760 +0100
@@ -142,6 +142,8 @@
 #endif /* KDKBDREP */
 }
 
+#undef rate
+
 static int
 KIOCSRATE_ioctl_ok(int rate, int delay) {
 #ifdef KIOCSRATE
@@ -168,8 +170,6 @@
 #endif /* KIOCSRATE */
 }
 
-#undef rate
-
 static void
 SetKbdRepeat(InputInfoPtr pInfo, char rad)
 {
